---
layout: page
permalink: /cv
title: CV
---

Date: March 2021

Summary
---------

I am a Senior Lecturer ('Førstelektor') at the Department of Biosciences, University of Oslo. I am also head of education and training for the Centre for Bioinformatics, Dept. of Informatics, Univ. of Oslo.

At the Dept. of Biosciences, I lead the implementation of Computing in Science Education in the bachelor degree. As part of that I organise the first-semester course "Introduction to Computational Modelling for the Biosciences" (BIOS1100), teaching new Bioscience students mathematical modeling of biological problems using Python.

For the Centre for Bioinformatics, I develop and coordinate a portfolio of credited courses as well as workshops for master and PhD students and other staff.

I am a self-taught bioinformatician with a firm background in biology. My research at the Centre for Ecological and Evolutionary Synthesis ([CEES](https://www.mn.uio.no/cees/english/)) at the Dept. of Biosciences is on new methods to represent (reference) genomic information. I previously had a central role in the sequencing and assembly of the Atlantic cod genome. I lead the Advisory Council for eInfrastructure ([fagrådet for eInfrastruktur](www.uio.no/for-ansatte/nettverk-moter/los-enhetene/forskningsinfrastruktur/einfrastruktur/)) at UiO.

I am a Software Carpentry instructor, teaching computational lab skills to scientists in the Nordic countries. I am an advocate of open source software, and reproducible computational biology. I share my work not only through publications, but also on my science blog at [lexnederbragt.com](https://lexnederbragt.com). My presentations can be found on [figshare](figshare.com/collections/Slides_of_my_talks_presentations/3488832) and [slideshare](http://slideshare.com/flxlex), and I am active on twitter ([@lexnederbragt](https://twitter.com/lexnederbragt)).

I have an analytical mind that loves problem solving. I value honesty and professionalism, and dislike hype. I believe that sharing and openness will further science more than competition. I choose impact over impact factor.


Research and Professional Experience
------------------------------------
* 2018- Senior lecturer, Dept. of Biosciences and Dept of Informatics, University of Oslo, Norway  
* 2018- Head of education and training, Centre for Bioinformatics, Dept of Informatics, University of Oslo, Norway  
* 2013-2018 Associate Professor (20%), Dept of Informatics, University of Oslo, Norway  
* 2005-2018  Postdoct. Res. Fellow, Researcher, Senior Engineer, Dept. of Biosciences, Univ. of Oslo, Norway  
* 2003-2005 Senior Scientist, Lingvitae AS, Oslo, Norway  
* 2002-2003 Postdoctoral research project University of Hawaii, USA  

Education
---------
* 2019-2020 Foundation programme Teaching and Learning in Higher Education ('Pedagogisk basiskompetanse')
* 1997-2001 PhD Biology, Utrecht Grad. School of Developmental Biology, Utrecht University, Netherlands  
* 1990-1996 BSc & MSc Biology, Utrecht University, Netherlands  

Leadership training
-------------------
Research Leadership ('Forskerledelse') for temporarily employed researchers at the Biological Institute, 8 half days. Organiser: Storyboard/Stride.no (2009)  
Research management for Phd students, NIBI (Netherlands Institute for Biology), (2000)

Teaching and teaching material
--------

**Courses organised**

BIOS1100 - Introduction to Computational Modelling in the Biosciences: <http://www.uio.no/studier/emner/matnat/ibv/BIOS1100>; new first-year bachelor course that I am coordinator and main teacher for; the course is the first implementation of the 'Computing in Science Education' project for the Biosciences study program, teaching python programming and modelling of real biological phenomena, 2017-

INF-BIO5121/9121 - High Throughput Sequencing technologies and bioinformatics analysis Univ. of Oslo: <http://inf-biox121.readthedocs.org/en/2015/>; I organise the course, coordinating with the five other teachers, and teach the assembly module: <http://inf-biox121.readthedocs.org/en/2016/Assembly/>, 2012-2016

**Contributions to university courses**

MBV-INF4410/9410 Bioinformatics for Molecular Biology (Univ. of Oslo): lecture on "The bioinformatics of sequencing and assembling genomes", and "What does it mean to do bioinformatics?", 2013-2016

BIO9905MERG1 - Bioinformatics for Metagenomic Analyses and Environmental Sequencing (Univ. of Oslo): lectures on "Next Generation Sequencing techniques and data relevant for metagenomics analyses" and "Assembly of metagenomes", 2011

BIO2120 Evolusjonsbiologi (Univ. of Oslo): lectures on "Evolution and Development" and "Evolution of Genes and Genomes"; group work, 2006 - 2007

Erasmus ICP course Marine Cell Biology (Observatoire Oceanologique, Banyuls-sur-mer, France): lectures on "Fundamental aspects of development" and "Cell cycle changes during development", 2000

**Workshops**  
[Workshop on Genome Assembly](https://github.com/lexnederbragt/denovo-assembly-tutorial) for the Norwegian Sequencing Centre course: [High Through-put Sequencing: technology basics, applications and bioinformatic analysis](https://www.sequencing.uio.no/events/2011/nsc_course_2011.html), 2011

**Contributions to workshops**  
[Next-Gen Sequence Analysis Workshop](http://angus.readthedocs.org/en/2015/week3.html) 'week 3' (intermediate and advanced skills) (invited), Michigan State University, 2015  
[Univ. California Davis Assembly Masterclass](http://davis-assembly-masterclass-2013.readthedocs.org/en/latest) (invited), 2013  
*De novo* genome assembly (invited), Univ. of Gothenburg, 2011


### Software and Data Carpentry
[Software Carpentry](http://software-carpentry.org/) and [Data Carpentry](http://www.datacarpentry.org/) are teaching researchers in science, engineering, medicine, and related disciplines worldwide computing and data analysis skills through short workshops using volunteer instructors.  I am a certified instructor of and organize and lead workshops in Norway and Sweden. Together with Karin Lagesen and with support from the Science Library (UiO), we have established Software and Data Carpentry at the University of Oslo, see [uio.no/carpentry](http://uio.no/carpentry), organising workshops more and more frequently, and increasing the number of certified instructors. I am also a certified *instructor trainer*, which enables me to give workshops for researchers that want to become instructors for Software or Data Carpentry.

**Software Carpentry workshops taught**  
University of Oslo - online: 2020, 2021  
University of Oslo: 2012, 2013, 2015 - 2019  
Netherlands eScience Centre: 2017  
University of Bergen: 2014  
Science for Life Laboratory, Stockholm, Sweden 2014  

**Carpentries Instructor Training workshops taught**  
University of Oslo: 2016, 2018  
The Carpentries - online: 2016-2019

**Software/Data Carpentry instructor retreat**  
I gave a session on Interactive learning techniques, recording: <https://www.youtube.com/watch?v=QptHJgzooU0>.

Grants
------

* Project "Integration of Jupyter and Canvas for digital assessment", financed by Link - Centre for Learning, Innovation & Academic Development, UiO, 2019
* Ratatosk Course Mobility Grant "Bringing Software Carpentry basic computing skills training to more Nordic countries" Nordic eInfrastructure Collaboration (NeIC), 2018

<!--
Listed in https://wiki.neic.no/w/ext/img_auth.php/c/c3/191220-Ratatosk-Final_report.pdf
-->

## Grants that I am listed on as collaborator

### European Research council
ERC Consolidator Grant: **CICHLIDX**, an integrative approach towards the understanding of an adaptive radiation of East African cichlid fishes (PI: Walter Salzburger, Univ. of Basel)

### Research Council of Norway
**REPEAT: Evolutionary and functional importance of simple repeats in
the genome** 'Toppforsk' project (PI: Kjetill Jakobsen)  
**Aqua Genome Project** (PI: Kjetill Jakobsen)  
**Norwegian Sequencing Centre** Research Infrastructure, phase II (PI: Kjetill Jakobsen, Univ. of Oslo)  
**CodReseq**: Translating the cod genome for aquaculture (PI: Kjetill Jakobsen)  
**Haddoc**, assessment of long-term effects of oil exposure on early life stages of Atlantic haddock (PI: Sonic Meier, Inst. of Marine Research)  
**Genome dynamics in early eukaryotic evolution**: importance of enigmatic lineages (PI: Russel Orr, Univ. of Oslo)  
**The genomic and physiological basis of invasiveness in a harmful house-invader** (InHouse) (PI: Inger Skrede, Univ. of Oslo)  

Selected blog posts
-------------------
"A new 1st semester bachelor course 'Introduction to Computational Modelling for the Biosciences'" <https://lexnederbragt.com/blog/2017-03-08-a-new-1st-semester-bachelor-course-introduction-to-computational-modelling-for-the-biosciences>

"Developments in high throughput sequencing – July 2016 edition" <https://lexnederbragt.com/blog/2016-07-08-developments-in-high-throughput-sequencing-july-2016-edition/>

"Active learning strategies for bioinformatics teaching" (2015)
<https://lexnederbragt.com/blog/2015-08-31-active-learning-strategies-for-bioinformatics-teaching-2/>

"On graph-based representations of a (set of) genomes" (2015) <https://lexnederbragt.com/blog/2015-04-09-on-graph-based-representations-of-a-set-of-genomes/>

"On the benefits of ‘open’ for teaching" (2014)
<https://lexnederbragt.com/blog/2014-11-04-on-the-benefits-of-open-for-teaching/>


Software
--------

Github: <https://github.com/lexnederbragt>

*Selected repositories:*  
[canvasCourseCli](https://github.com/lexnederbragt/canvasCourseCli): command line tool for retrieving, adding and updating content, pages and files exposed to students and teachers, for a Canvas course.  
[SequenceTools](https://github.com/lexnederbragt/sequencetools): scripts to work with DNA sequence and related files.  
[RepSeq](https://github.com/lexnederbragt/RepSeq): estimating copy numbers of assembled DNA pieces (contigs), Nederbragt et al 2010.  

Invited Seminars
----------------
Slide decks:

* <http://www.figshare.com/collections/Slides_of_my_talks_presentations/3488832>
* <http://www.slideshare.net/flxlex>

Selected presentations:

"Genomic variation detection in Atlantic cod using long reads", SMRT Scientific Symposium, Leiden, The Netherlands, 2016 <https://figshare.com/articles/Genomic_variation_detection_in_Atlantic_cod_using_long_reads/3413887>

"Local, central or cloud: why we chose to build our e-infrastructure for sequence analysis embedded in the university of Oslo HPC cluster" workshop "e-infrastructures for Massively Parallel Sequencing", Uppsala, 2015 <https://www.youtube.com/watch?v=g5JtEQPEQiI>

"Coding & Best Practice in Programming: Why it matters so much in the NGS era", SeqAhead Scientific Meeting 2014 "NGS Data after the Gold Rush", Norwich, 2014 <http://www.slideshare.net/flxlex/140507-seq-aheadngscodingbestpractices>

"Experiences with Combining PacBio with Short Read Technology for Improved de novo Assembly of Complex Genomes", 4th Next Generation Sequencing Congress, London 2012 <http://www.slideshare.net/flxlex/131014-cees-holmenreproducibility>

Professional and Community Service
----------------------------------
**Grant Panels and Reviewing**  
Leader, The Resource Allocation Committee for CPU and storage access at National HPC facilities (UNINETT Sigma2 AS), Norway (member 2014-2018, leader 2019-)

**Advisory boards**  
Lederteam Center for Computing in Science Education (CCSE), UiO, 2017-  
Executive Council 'The Carpentries' (https://carpentries.org), Secretary (2018), Chair (2021), 2018-2021
Leader, Advisory Council ('fagråd') eInfrastructure, University of Oslo (2016-)  
Leader of the board, Digital Life Norway Research School (2016-2020)  
MicrobesNG Microbial Genome Sequencing service and Open Strain Archive, Univ. of Birmingham, External Advisory Board (2014-2019)  

<!-- Member, Council for Research ('fagutvalg for forskning'), Unit – The Norwegian Directorate for ICT (2018-)  -->
<!-- Member, task force 'Action plan Open Science Norway', Universities Norway ('Universitets- og høgskolerådet', UHR) (2019-)  -->

**Manuscript Reviewing**  
Open reviews: Publons <https://publons.com/author/18712/alexander-j-nederbragt>

Reviewer for Nature Methods, Nature Scientific Data, Bioinformatics, BMC Bioinformatics, BMC Genomics, Animal Genetics, Gigascience

Research Mentoring
------------------
**Graduate Students**  
Co-supervisor PhD students, Univ. of Oslo since 2008, 7 finished

**Undergraduate Researchers**  
Main supervisor MSc students, Univ. of Oslo since 2014, 5 finished
Co-supervisor MSc students, Univ. of Oslo since 2013, 6 finished

Volunteer activities
-------------------
Leader, 'kodeklubben Bærum' (code club), offering coding classes to children from 8 to 14 (2016-)  
Board member of the residential condominium 'Hiltonåsen', Slependen, Norway 2009-2015  
Board member of choir 'Saudade' Utrecht, The Netherlands, 2000-2001  
Board member of student rowing club 'A.U.S.R. Orca', Utrecht, The Netherlands, 1994 - 1999  

Languages
---------

Dutch: first language, excellent oral/written  
English: excellent oral/written  
Norwegian: good oral/written  
German: conversational


List of publications
--------------------

See also my [Google Scholar profile](https://scholar.google.no/citations?user=lSNc07gAAAAJ&hl=en) or [ORCID page](http://orcid.org/0000-0001-5539-0999).

# Preprints


Khelik, K., **Nederbragt, A. J.**, Sandve, G. K., & Rognes, T. (2018).
NucMerge: Genome assembly quality improvement assisted by alternative
assemblies and paired-end Illumina reads. *bioRxiv*, 483701.
<https://doi.org/10.1101/483701>


# Peer-reviewed publications


Grytten, I., Rand, K. D., **Nederbragt, A. J.**, & Sandve, G. K. (2020).
Assessing graph-based read mappers against a baseline approach
highlights strengths and weaknesses of current methods. *BMC Genomics*,
*21*(1), 282. <https://doi.org/10.1186/s12864-020-6685-y>



Khelik, K., Sandve, G. K., **Nederbragt, A. J.**, & Rognes, T. (2020).
NucBreak: Location of structural errors in a genome assembly by using
paired-end Illumina reads. *BMC Bioinformatics*, *21*(1), 66.
<https://doi.org/10.1186/s12859-020-3414-0>



Nederbragt, A., Harris, R. M., Hill, A. P., & Wilson, G. (2020). Ten
quick tips for teaching with participatory live coding. *PLOS
Computational Biology*, *16*(9), e1008090.
<https://doi.org/10.1371/journal.pcbi.1008090>



Grytten, I., Rand, K. D., **Nederbragt, A. J.**, Storvik, G. O., Glad, I.
K., & Sandve, G. K. (2019). Graph Peak Caller: Calling ChIP-seq peaks on
graph-based reference genomes. *PLOS Computational Biology*, *15*(2),
e1006731. <https://doi.org/10.1371/journal.pcbi.1006731>



Jørgensen, T. E., Karlsen, B. O., Emblem, Å., Breines, R., Andreassen,
M., Rounge, T. B., **Nederbragt, A. J.**, Jakobsen, K. S., Nymark, M.,
Ursvik, A., Coucheron, D. H., Jakt, L. M., Nordeide, J. T., Moum, T., &
Johansen, S. D. (2018). Mitochondrial genome variation of Atlantic cod.
*BMC Research Notes*, *11*, 397.
<https://doi.org/10.1186/s13104-018-3506-3>



Lie, K. K., Tørresen, O. K., Solbakken, M. H., Rønnestad, I.,
Tooming-Klunderud, A., **Nederbragt, A. J.**, Jentoft, S., & Sæle, Ø.
(2018). Loss of stomach, loss of appetite? Sequencing of the ballan
wrasse (Labrus bergylta) genome and intestinal transcriptomic profiling
illuminate the evolution of loss of stomach function in fish. *BMC
Genomics*, *19*, 186. <https://doi.org/10.1186/s12864-018-4570-8>



Tørresen, O. K., Brieuc, M. S. O., Solbakken, M. H., Sørhus, E.,
**Nederbragt, A. J.**, Jakobsen, K. S., Meier, S., Edvardsen, R. B., &
Jentoft, S. (2018). Genomic architecture of haddock (Melanogrammus
aeglefinus) shows expansions of innate immune genes and short tandem
repeats. *BMC Genomics*, *19*.
<https://doi.org/10.1186/s12864-018-4616-y>



Varadharajan, S., Sandve, S. R., Gillard, G. B., Tørresen, O. K.,
Mulugeta, T. D., Hvidsten, T. R., Lien, S., Asbjørn Vøllestad, L.,
Jentoft, S., **Nederbragt, A. J.**, & Jakobsen, K. S. (2018). The Grayling
Genome Reveals Selection on Gene Expression Regulation after
Whole-Genome Duplication. *Genome Biology and Evolution*, *10*(10),
2785–2800. <https://doi.org/10.1093/gbe/evy201>



Elgvin, T. O., Trier, C. N., Tørresen, O. K., Hagen, I. J., Lien, S.,
**Nederbragt, A. J.**, Ravinet, M., Jensen, H., & Sætre, G.-P. (2017). The
genomic mosaicism of hybrid speciation. *Science Advances*, *3*(6),
e1602996. <https://doi.org/10.1126/sciadv.1602996>



Khelik, K., Lagesen, K., Sandve, G. K., Rognes, T., & **Nederbragt, A. J.**
(2017). NucDiff: In-depth characterization and annotation of differences
between two sets of DNA sequences. *BMC Bioinformatics*, *18*, 338.
<https://doi.org/10.1186/s12859-017-1748-z>



Rand, K. D., Grytten, I., **Nederbragt, A. J.**, Storvik, G. O., Glad, I.
K., & Sandve, G. K. (2017). Coordinates and intervals in graph-based
reference genomes. *BMC Bioinformatics*, *18*, 263.
<https://doi.org/10.1186/s12859-017-1678-9>



Simovski, B., Vodak, D., Gundersen, S., Domanska, D., Abdulrahman Azab,
Holden, L., Holden, M., Grytten, I., Rand, K., Drablos, F., Johansen,
M., Mora, A., Lund-Andersen, C., Fromm, B., Eskeland, R., Gabrielsen, O.
S., Ferkingstad, E., Sigve Nakken, Bengtsen, M., **Nederbragt, A. J.**,
Thorarensen, H. S., Akse, J. A., Glad, I., Hovig, E., & Sandve, G. K.
(2017). Gsuite HyperBrowser Version2.0b. GigaScience Database.
<https://doi.org/10.5524/100292>



Simovski, B., Vodák, D., Gundersen, S., Domanska, D., Azab, A., Holden,
L., Holden, M., Grytten, I., Rand, K., Drabløs, F., Johansen, M., Mora,
A., Lund-Andersen, C., Fromm, B., Eskeland, R., Gabrielsen, O. S.,
Ferkingstad, E., Nakken, S., Bengtsen, M., **Nederbragt, A. J.**, Sif
Thorarensen, H., Andreas Akse, J., Glad, I., Hovig, E., & Sandve, G. K.
(2017). GSuite HyperBrowser: Integrative analysis of dataset collections
across the genome and epigenome. *GigaScience*.
<https://doi.org/10.1093/gigascience/gix032>



Tørresen, O. K., Star, B., Jentoft, S., Reinar, W. B., Grove, H.,
Miller, J. R., Walenz, B. P., Knight, J., Ekholm, J. M., Peluso, P.,
Edvardsen, R. B., Tooming-Klunderud, A., Skage, M., Lien, S., Jakobsen,
K. S., & **Nederbragt, A. J.** (2017). An improved genome assembly uncovers
prolific tandem repeats in Atlantic cod. *BMC Genomics*, *18*(1), 95.
<https://doi.org/10.1186/s12864-016-3448-x>



Wilson, G., Bryan, J., Cranston, K., Kitzes, J., **Nederbragt, L.**, & Teal,
T. K. (2017). Good enough practices in scientific computing. *PLOS
Computational Biology*, *13*(6), e1005510.
<https://doi.org/10.1371/journal.pcbi.1005510>



Frank, J. A., Pan, Y., Tooming-Klunderud, A., Eijsink, V. G. H.,
McHardy, A. C., **Nederbragt, A. J.**, & Pope, P. B. (2016). Improved
metagenome assemblies and taxonomic binning using long-read circular
consensus sequence data. *Scientific Reports*, *6*, 25373.
<https://doi.org/10.1038/srep25373>



Lien, S., Koop, B. F., Sandve, S. R., Miller, J. R., Kent, M. P., Nome,
T., Hvidsten, T. R., Leong, J. S., Minkley, D. R., Zimin, A., Grammes,
F., Grove, H., Gjuvsland, A., Walenz, B., Hermansen, R. A., von
Schalburg, K., Rondeau, E. B., Di Genova, A., Samy, J. K. A., Olav Vik,
J., Vigeland, M. D., Caler, L., Grimholt, U., Jentoft, S., Inge Våge,
D., de Jong, P., Moen, T., Baranski, M., Palti, Y., Smith, D. R., Yorke,
J. A., **Nederbragt, A. J.**, Tooming-Klunderud, A., Jakobsen, K. S., Jiang,
X., Fan, D., Hu, Y., Liberles, D. A., Vidal, R., Iturra, P., Jones, S.
J. M., Jonassen, I., Maass, A., Omholt, S. W., & Davidson, W. S. (2016).
The Atlantic salmon genome provides insights into rediploidization.
*Nature*, *533*(7602), 200–205. <https://doi.org/10.1038/nature17164>



Malmstrøm, M., Matschiner, M., Tørresen, O. K., Star, B., Snipen, L. G.,
Hansen, T. F., Baalsrud, H. T., **Nederbragt, A. J.**, Hanel, R.,
Salzburger, W., Stenseth, N. C., Jakobsen, K. S., & Jentoft, S. (2016).
Evolution of the immune system influences speciation rates in teleost
fishes. *Nature Genetics*, *48*(10), 1204–1210.
<https://doi.org/10.1038/ng.3645>



Solbakken, M. H., Tørresen, O. K., **Nederbragt, A. J.**, Seppola, M.,
Gregers, T. F., Jakobsen, K. S., & Jentoft, S. (2016). Evolutionary
redesign of the Atlantic cod (*Gadus* *Morhua* L.) Toll-like receptor
repertoire by gene losses and expansions. *Scientific Reports*, *6*,
25211. <https://doi.org/10.1038/srep25211>



Star, B., Tørresen, O. K., **Nederbragt, A. J.**, Jakobsen, K. S.,
Pampoulie, C., & Jentoft, S. (2016). Genomic characterization of the
Atlantic cod sex-locus. *Scientific Reports*, *6*, 31235.
<https://doi.org/10.1038/srep31235>



Brembu, T., Winge, P., Tooming-Klunderud, A., **Nederbragt, A. J.**,
Jakobsen, K. S., & Bones, A. M. (2014). The chloroplast genome of the
diatom Seminavis robusta: New features introduced through multiple
mechanisms of horizontal gene transfer. *Marine Genomics*, *16*, 17–27.
<https://doi.org/10.1016/j.margen.2013.12.002>



Hermansen, J. S., Haas, F., Trier, C. N., Bailey, R. I., Nederbragt, A.
J., Marzal, A., & Sætre, G.-P. (2014). Hybrid speciation through sorting
of parental incompatibilities in Italian sparrows. *Molecular Ecology*,
*23*(23), 5831–5842. <https://doi.org/10.1111/mec.12910>



**Nederbragt, A. J.** (2014). On the middle ground between open source and
commercial software - the case of the Newbler program. *Genome Biology*,
*15*(4), 113. <https://doi.org/10.1186/gb4173>



Röblitz, T., Saastad, O. W., Eide, H. A., Michalickova, K., Nederbragt,
A. J., & Star, B. (2014). *Scaling of Biological Data Work ows to Large
HPC Systems - A Case Study in Marine Genomics -*. Zenodo.
<https://doi.org/10.5281/zenodo.823031>



Siddiqui, H., Lagesen, K., **Nederbragt, A. J.**, Eri, L. M., Jeansson, S.
L., & Jakobsen, K. S. (2014). Pathogens in Urine from a Female Patient
with Overactive Bladder Syndrome Detected by Culture-independent High
Throughput Sequencing: A Case Report. *The Open Microbiology Journal*,
*8*(1). <https://doi.org/10.2174/1874285801408010148>



Star, B., **Nederbragt, A. J.**, Hansen, M. H. S., Skage, M., Gilfillan, G.
D., Bradbury, I. R., Pampoulie, C., Stenseth, N. C., Jakobsen, K. S., &
Jentoft, S. (2014). Palindromic Sequence Artifacts Generated during Next
Generation Sequencing Library Preparation from Historic and Ancient DNA.
*PLOS ONE*, *9*(3), e89676.
<https://doi.org/10.1371/journal.pone.0089676>



Tooming-Klunderud, A., Sogge, H., Rounge, T. B., **Nederbragt, A. J.**,
Lagesen, K., Glöckner, G., Hayes, P. K., Rohrlack, T., & Jakobsen, K. S.
(2013). From Green to Red: Horizontal Gene Transfer of the Phycoerythrin
Gene Cluster between Planktothrix Strains. *Appl. Environ. Microbiol.*,
*79*(21), 6803–6812. <https://doi.org/10.1128/AEM.01455-13>



Espelund, M., Minge, M. A., Gabrielsen, T. M., **Nederbragt, A. J.**,
Shalchian-Tabrizi, K., Otis, C., Turmel, M., Lemieux, C., & Jakobsen, K.
S. (2012). Genome Fragmentation Is Not Confined to the Peridinin Plastid
in Dinoflagellates. *PLOS ONE*, *7*(6), e38809.
<https://doi.org/10.1371/journal.pone.0038809>



Logares, R., Haverkamp, T. H. A., Kumar, S., Lanzén, A., Nederbragt, A.
J., Quince, C., & Kauserud, H. (2012). Environmental microbiology
through the lens of high-throughput DNA sequencing: Synopsis of current
platforms and bioinformatics approaches. *Journal of Microbiological
Methods*, *91*(1), 106–113.
<https://doi.org/10.1016/j.mimet.2012.07.017>



Siddiqui, H., Lagesen, K., **Nederbragt, A. J.**, Jeansson, S. L., &
Jakobsen, K. S. (2012). Alterations of microbiota in urine from women
with interstitial cystitis. *BMC Microbiology*, *12*(1), 205.
<https://doi.org/10.1186/1471-2180-12-205>



Brede, D. A., Snipen, L. G., Ussery, D. W., **Nederbragt, A. J.**, & Nes, I.
F. (2011). Complete Genome Sequence of the Commensal Enterococcus
faecalis 62, Isolated from a Healthy Norwegian Infant. *Journal of
Bacteriology*, *193*(9), 2377–2378.
<https://doi.org/10.1128/JB.00183-11>



Gabrielsen, T. M., Minge, M. A., Espelund, M., Tooming-Klunderud, A.,
Patil, V., **Nederbragt, A. J.**, Otis, C., Turmel, M., Shalchian-Tabrizi,
K., Lemieux, C., & Jakobsen, K. S. (2011). Genome Evolution of a
Tertiary Dinoflagellate Plastid. *PLOS ONE*, *6*(4), e19132.
<https://doi.org/10.1371/journal.pone.0019132>



Siddiqui, H., **Nederbragt, A. J.**, Lagesen, K., Jeansson, S. L., &
Jakobsen, K. S. (2011). Assessing diversity of the female urine
microbiota by high throughput sequencing of <span class="nocase">16S
rDNA</span> amplicons. *BMC Microbiology*, *11*(1), 244.
<https://doi.org/10.1186/1471-2180-11-244>



Star, B., **Nederbragt, A. J.**, Jentoft, S., Grimholt, U., Malmstrøm, M.,
Gregers, T. F., Rounge, T. B., Paulsen, J., Solbakken, M. H., Sharma,
A., Wetten, O. F., Lanzén, A., Winer, R., Knight, J., Vogel, J.-H.,
Aken, B., Andersen, Ø., Lagesen, K., Tooming-Klunderud, A., Edvardsen,
R. B., Tina, K. G., Espelund, M., Nepal, C., Previti, C., Karlsen, B.
O., Moum, T., Skage, M., Berg, P. R., Gjøen, T., Kuhl, H., Thorsen, J.,
Malde, K., Reinhardt, R., Du, L., Johansen, S. D., Searle, S., Lien, S.,
Nilsen, F., Jonassen, I., Omholt, S. W., Stenseth, N. C., & Jakobsen, K.
S. (2011). The genome sequence of Atlantic cod reveals a unique immune
system. *Nature*, *477*(7363), 207–210.
<https://doi.org/10.1038/nature10342>



Løvoll, M., Wiik-Nielsen, J., Grove, S., Wiik-Nielsen, C. R.,
Kristoffersen, A. B., Faller, R., Poppe, T., Jung, J., Pedamallu, C. S.,
**Nederbragt, A. J.**, Meyerson, M., Rimstad, E., & Tengs, T. (2010). A
novel totivirus and piscine reovirus (PRV) in Atlantic salmon (Salmo
salar) with cardiomyopathy syndrome (CMS). *Virology Journal*, *7*(1), 309. <https://doi.org/10.1186/1743-422X-7-309>



**Nederbragt, A. J.**, Rounge, T. B., Kausrud, K. L., & Jakobsen, K. S.
(2010). Identification and Quantification of Genomic Repeats and Sample
Contamination in Assemblies of 454 Pyrosequencing Reads. *Sequencing*.
Research Article,
https://www.hindawi.com/journals/sequencing/2010/782465/.
<https://doi.org/10.1155/2010/782465>



Wetten, O. F., **Nederbragt, A. J.**, Wilson, R. C., Jakobsen, K. S.,
Edvardsen, R. B., & Andersen, Ø. (2010). Genomic organization and gene
expression of the multiple globins in Atlantic cod: Conservation of
globin-flanking genes in chordates infers the origin of the vertebrate
globin clusters. *BMC Evolutionary Biology*, *10*(1), 315.
<https://doi.org/10.1186/1471-2148-10-315>



Johansen, S. D., Coucheron, D. H., Andreassen, M., Karlsen, B. O.,
Furmanek, T., Jørgensen, T. E., Emblem, Å., Breines, R., Nordeide, J.
T., Moum, T., **Nederbragt, A. J.**, Stenseth, N. C., & Jakobsen, K. S.
(2009). Large-scale sequence analyses of Atlantic cod. *New
Biotechnology*, *25*(5), 263–271.
<https://doi.org/10.1016/j.nbt.2009.03.014>



Rounge, T. B., Rohrlack, T., **Nederbragt, A. J.**, Kristensen, T., &
Jakobsen, K. S. (2009). A genome-wide analysis of nonribosomal peptide
synthetase gene clusters and their peptides in a Planktothrix rubescens
strain. *BMC Genomics*, *10*(1), 396.
<https://doi.org/10.1186/1471-2164-10-396>



Siddiqui, H., **Nederbragt, A. J.**, & Jakobsen, K. S. (2009). A solid-phase
method for preparing human DNA from urine for diagnostic purposes.
*Clinical Biochemistry*, *42*(10), 1128–1135.
<https://doi.org/10.1016/j.clinbiochem.2009.03.010>



**Nederbragt, A. J.**, Balasingham, A., Sirevåg, R., Utkilen, H., Jakobsen,
K. S., & Anderson-Glenna, M. J. (2008). Multiple-locus variable-number
tandem repeat analysis of Legionella pneumophila using multi-colored
capillary electrophoresis. *Journal of Microbiological Methods*,
*73*(2), 111–117. <https://doi.org/10.1016/j.mimet.2008.02.007>



Lespinet, O., **Nederbragt, A. J.**, Cassan, M., Dictus, W. J., van Loon, A.
E., & Adoutte, A. (2002). Characterisation of two snail genes in the
gastropod mollusc Patella vulgata. Implications for understanding the
ancestral function of the snail-related genes in Bilateria. *Development
Genes and Evolution*, *212*(4), 186–195.
<https://doi.org/10.1007/s00427-002-0228-1>



**Nederbragt, A. J.**, Lespinet, O., Wageningen, S. V., Loon, A. E. V.,
Adoutte, A., & Dictus, W. J. A. G. (2002). A lophotrochozoan twist gene
is expressed in the ectomesoderm of the gastropod mollusk Patella
vulgata. *Evolution & Development*, *4*(5), 334–343.
<https://doi.org/10.1046/j.1525-142X.2002.02020.x>



**Nederbragt, A. J.**, te Welscher, P., van den Driesche, S., van Loon, A.
E., & Dictus, W. J. (2002). Novel and conserved roles for
orthodenticle/otx and orthopedia/otp orthologs in the gastropod mollusc
Patella vulgata. *Development Genes and Evolution*, *212*(7), 330–337.
<https://doi.org/10.1007/s00427-002-0246-z>



**Nederbragt, A. J.**, van Loon, A. E., & Dictus, W. J. A. G. (2002a).
Evolutionary biology: *Hedgehog* crosses the snail’s midline. *Nature*,
*417*(6891), 811–812. <https://doi.org/10.1038/417811b>



**Nederbragt, A. J.**, van Loon, A. E., & Dictus, W. J. A. G. (2002b).
Expression of Patella vulgata Orthologs of engrailed and dpp-BMP2/4 in
Adjacent Domains during Molluscan Shell Development Suggests a Conserved
Compartment Boundary Mechanism. *Developmental Biology*, *246*(2),
341–355. <https://doi.org/10.1006/dbio.2002.0653>



van der Kooij, A., **Nederbragt, A. J.**, Goedemans, H. J., & van Loon,
AndréE. (1996). The stringlike genes of the limpet Patella vulgata.
*Gene*, *172*(2), 261–265.
<https://doi.org/10.1016/0378-1119(96)00164-3>


# Conference Contributions


Eliassen, J. E., Bøe, M. V., Nederbragt, L., & Gregers, T. F. (2021).
Motivasjon for beregningsorientert biologi og sammenhengen med
matematikk R2 fra videregående opplæring. *Nordic Journal of STEM
Education*, *5*(1). <https://doi.org/10.5324/njsteme.v5i1.3917>



Gregers, T. F., & Nederbragt, L. (2019). <span
class="nocase">Lektorstudenter utvikler unik kompetanse og bidrar til
økt kvalitet på begynneremner gjennom en undervisningsrettet
master</span>. *Nordic Journal of STEM Education*, *3*(1), 23–27.
<https://doi.org/10.5324/njsteme.v3i1.2992>


# Book chapters


Tørresen, O. K., Rise, M. L., Jin, X., Star, B., MacKenzie, S.,
Jakobsen, K. S., **Nederbragt, A. J.**, & Jentoft, S. (2016). 3 - An
improved version of the Atlantic cod genome and advancements in
functional genomics: Implications for the future of cod farming. In
*Genomics in Aquaculture* (pp. 45–72). San Diego: Academic Press.
<https://doi.org/10.1016/B978-0-12-801418-9.00003-2>



Tørresen, O. K., Star, B., Jentoft, S., Jakobsen, K. S., & Nederbragt,
A. J. (2016). 1 - The new era of genome sequencing using high-throughput
sequencing technology: Generation of the first version of the Atlantic
cod genome. In *Genomics in Aquaculture* (pp. 1–20). San Diego: Academic
Press. <https://doi.org/10.1016/B978-0-12-801418-9.00001-9>


# Thesis


**Nederbragt, A. J.** (2002). *<span class="nocase">The evolution of
developmental programs : a case study in the gastropod mollusc Patella
vulgata</span>* (Dissertation). Retrieved from
<http://dspace.library.uu.nl/handle/1874/213>


# Software publications


Crusoe, M. R., Alameldin, H. F., Awad, S., Boucher, E., Caldwell, A.,
Cartwright, R., Charbonneau, A., Constantinides, B., Edvenson, G., Fay,
S., Fenton, J., Fenzl, T., Fish, J., Garcia-Gutierrez, L., Garland, P.,
Gluck, J., González, I., Guermond, S., Guo, J., Gupta, A., Herr, J. R.,
Howe, A., Hyer, A., Härpfer, A., Irber, L., Kidd, R., Lin, D., Lippi,
J., Mansour, T., McA’Nulty, P., McDonald, E., Mizzi, J., Murray, K. D.,
Nahum, J. R., Nanlohy, K., **Nederbragt, A. J**., Ortiz-Zuazaga, H., Ory,
J., Pell, J., Pepe-Ranney, C., Russ, Z. N., Schwarz, E., Scott, C.,
Seaman, J., Sievert, S., Simpson, J., Skennerton, C. T., Spencer, J.,
Srinivasan, R., Standage, D., Stapleton, J. A., Steinman, S. R., Stein,
J., Taylor, B., Trimble, W., Wiencko, H. L., Wright, M., Wyss, B.,
Zhang, Q., zyme, en, & Brown, C. T. (2015). The khmer software package:
Enabling efficient nucleotide sequence analysis. *F1000Research*.
<https://doi.org/10.12688/f1000research.6924.1>


# Published lesson materials


Becker, E. A., Koch, C., Word, K., Harris, R. M., Sane, M., Nederbragt,
L., Michonneau, F., Jordan, K. L., Hodge, A. E., Erdmann, C., Rokem, A.,
Laderas, T., Wilson, G., Lee, I., Cadzow, M., Brown, S., Hodges, T.,
Nenadic, A., Duckles, J., Rampin, R., Teal, T., Leinweber, K., Martinez,
P. A., Emonet, R., Konovalov, A., Jankowski, E., Capes, G., Williams,
J., Koziar, K., Palopoli, N., Spies, N., Weber, A., amoskane, Schürch,
A., AnnaWilliford, Brian Ballsun-Stanton, Oliver, J., Black, M.,
Kindlon, N., Guillou, S., Yehudi, Y., Walsh, A. J., Collier, A. B.,
Switzer, C., Brueffer, C., Morris, C., Allan, D., DanielBrett, Quinn,
D., Vanichkina, D., Jennings, D., davidbenncsiro, Williamson, E. P.,
Gates, J. M., Atzberger, J., Bradley, J., Pellman, J., Bradley, J.,
Cranston, K., Kees den Heijer, Laurence, Marie-Helene Burle, Henry, M.,
naught101, Davis, N., Tierney, N., Petraea, Brown, S. M., Stevens, S.,
satya-vinay, Sean, Kiburu, S. A. N., Helfrich, S., Moss, S., & Pereira,
T. M. D. (2019). The Carpentries Instructor Training June 2019.
*Zenodo*. <https://doi.org/10.5281/zenodo.3258398>



Ahmadia, A., Allen, J., Banaszkiewicz, P., Becker, E., Bekolay, T.,
Belkin, M., Blischak, J., Boughton, A., Bray, E., Cabunoc Mayes, A.,
Carroll, I., Crouch, S., Davis, M., Davis, N., Devenyi, G. A., Duckles,
J., Emonet, R., Fortin, F.-A., Gonzalez, I., Hamm, C., Hansen, M.,
Harris, R. M., Henninger, F., Hertweck, K., Hinsen, K., Hodge, A.,
Jackson, M., King, W. T., Kitzes, J., Koch, C., Lang, D., Liversidge,
T., Michonneau, F., Mills, B., **Nederbragt, L.**, Nelson, E., O’Leary, A.,
Ortiz-Zuazaga, H., Patitsas, E., Pawlik, A., Perez, F., Pipitone, J.,
Poisot, T., Rokem, A., Silva, R., Steyn, J., Teal, T., Tröndle, T.,
Tweedie, F., van der Walt, A., Vie, J.-J., Walker, J., Walsh, A.,
Weaver, B., White, E., Wilkerson, C., Williams, J., Williamson, E. P.,
Wilson, G., & Woo, K. (2017, February). Software Carpentry: Instructor
Training. <https://doi.org/10.5281/zenodo.278229>



Achterberg, H., Adams, J., Adelman, J., Allen, J., Aranda, J., Bae, S.,
Banaszkiewicz, P., Barmby, P., Barr, E., Beitey, D., Bekolay, T.,
Berghold, J., Bjørnstad, M., Blischak, J., Boissonneault, M., Bosley,
C., Bostroem, A., Boughton, A., Brase, R., Brown, A., Brown, K., Cabunoc
Mayes, A., Carroll, I., Chase, J., Childs, S., Choi, J., Clare, R.,
Clayton, S., Cock, P., Connell, M., Corvellec, M., Coudrat, T., Dale,
R., Davis, M., Davis, N., Davison, A., Demichelis, R., Devenyi, G. A.,
Dolson, E., Dotson, D., Duchesne, L., Duckles, J., Emonet, R., Endsley,
K. A., Fauchereau, N., Ford, T., Förstner, K., Gonzalez, I., Gosmann,
J., Gosset, J., Gray, J., Greshake, B., Guillou, S., Haley, M., Hames,
S., Hamrick, J. B., Hannon, E., Hansen, M., Hertweck, K., Hinsen, K.,
Hjelm, J., Hodges, T., Howard, D., Irving, D., Jackson, M., Jolly, B.,
Jones, N., Joyce, B., Ketcheson, D., King, W. T., Klerke, S., Ko, L.,
Kwong, A., Laken, B., Lapp, H., Latornell, D., Leeman, J. R., Leyder,
J.-C., Ligtenberg, W., Lin, J., Lonsdale, A., Manhaes Savio, A., May,
R., Mazur, D., Michonneau, F., Mills, B., Mughal, Z., **Nederbragt, L.**,
Neufeld, R., O’Leary, A., Obeng, A., Ory, J., Osiecka, N., Pipitone, J.,
Pinska, A., Poisot, T., Pomorski, P., Raghupathy, N., Rathgeber, F.,
Rawls, M., Richie-Halford, A., Riley, J., Robitaille, T., Rokem, A.,
Roswell, M., Sadjadi, M., Sales de Andrade, E., Schmeier, S., Sheneman,
L., Shiffer, A., Shkurti, A., Silva, R., Soranzo, N., Spence, E.,
Srinath, A., Staneva, V., Stapleton, J., Stucky, B., Taylor, C., Teal,
T., Telenczuk, B., Thomas, I., Thorne, B., Torres, G., Tröndle, T.,
Upani, J., Vahtras, O., van der Helm, E., Walsh, A., Walter, N.,
Weitzel, D., Wheeler, D., White, E., Williamson, E. P., Willmore, F.,
Wilson, A., Wilson, G., Xiao, X., & Zonca, A. (2017). Software
Carpentry: Programming with Python. *Zenodo*.
<https://doi.org/10.5281/zenodo.278222>



Ahmadia, A., Allen, J., Appling, A., Aubin, S., Bachant, P., Baird, D.,
Banaszkiewicz, P., Barmby, P., Batut, B., Bekolay, T., Belkin, M.,
Blischak, J., Bonsma, M., Borrelli, J., Boughton, A., Bouquin, D.,
Brauning, R., Brett, M., Brown, A., Cabunoc Mayes, A., Charlesworth, J.,
Charlton, B., Chen, D., Christensen, G., Collings, R., Corvellec, M.,
Davis, M., Devenyi, G. A., Dolson, E., Duchesne, L., Duckles, J.,
Emonet, R., Esteve, L., Farsarakis, E., Fauber, B., Fouilloux, A.,
Forstner, K., Geiger, S., Gonzalez, I., Guarinello, M., Guillou, S.,
Hadwin, J., Haeni, M., Haessig, P., Hannah, N., Hansen, M.,
Harihareswara, S., Heilmaier, A., Heroux, M., Hertweck, K., Hilboll, A.,
Hinsen, K., Huang, D., Ismiraldi, Y., Jackson, M., Jacobs, C., Jarecka,
D., Johnston, L. W., Jones, D., Jedrzejewski-Szmek, Z., Kelly, T., King,
W. T., Kluyver, T., Konrad, B., Kuzak, M., Küderle, A., Labrie, K.,
Lapp, H., Latornell, D., Laufersweiler, M., LeBauer, D., Lee, K.,
Liffers, M., Llebot, C., Loucks, C., Ma, K., Marwaha, K., May, R.,
Michonneau, F., Mills, B., Mueller, A., Munk, M., Nagraj, V.,
**Nederbragt, L.**, Nunez-Iglesias, J., O’Brien, B., O’Leary, A., Olsson,
C., Panitz, M., Pawsey, C., Pfenninger, S., Pipitone, J., Poisot, T.,
Preney, P., Rice, T., Riemer, K., Rio Deiros, D., Robinson, N.,
Rockenberger, A., Rohl, A., Rokem, A., Sacks, B., Sarahan, M., Schmeier,
S., Schmider, H., Shellito, P., Shriwise, P. C., Silva, R., Smithyman,
B., Soranzo, N., Steinbach, P., Stevens, S., Stueker, O., Stuermer, B.,
Timbers, T., Traphagen, D., Trondle, T., van der Walt, A., Vandervalk,
S., Watson, G., Weaver, B., Wheelhouse, M., White, E., Williamson, E.
P., Wilson, G., Wu, S., & Zhang, Q. (2017). Software Carpentry: Version
Control with Git. *Zenodo*. <https://doi.org/10.5281/zenodo.278219>



Aldazabal Mensa, I., Alexander, H., Allen, J., Alsheikh-Hussain, A.,
Attali, D., Baird, D., Baltzell, A., Banaszkiewicz, P., Barmby, P.,
Beagrie, R., Beane, G., Bedini, A., Bekolay, T., Belikov, E., Bell, J.,
Bill, B., Blin, K., Blischak, J., Boardman, S., Boissonneault, M.,
Bonnie, J., Boughton, A., Brase, R., Brown, A., Brunson, D., Bryk, J.,
Buske, O., Cabunoc Mayes, A., Carlise, M., Chen, D., Chung, K., Chung,
K., Deveau, D., Devenyi, G. A., Doehle, P., Dolson, E., Doyle, M.,
Duckles, J., Emonet, R., Eyers, D., Fernandes, F., Fontenelle, H.,
Gacenga, F., Gardner, P., Gidden, M., Gonzalez, I., Gray, N., Guillou,
S., Hagh, V. F., Hansen, M., Harstad, E., Hohenstein, T., Howe, A.,
Imamoglu, F., Irving, D., Jackson, M., Jane McTavish, E., Jennings, M.,
Jones, D., Jones, M. A., Keener, A., Keipert, K., Kelly, T., Kim, J. T.,
King, W. T., Koch, C., Konrad, B., Lake, S., Lang, D., Latornell, D.,
Lijnzaad, P., Ma, E., Mac, A., Madin, J., Magle, T., Mandel, M., Marini,
C., Marwaha, K., Mashchenko, S., McCloy, D., Michonneau, F., Middleton,
R., Milhans, J., Mills, B., Miotto, A., Mount, S., **Nederbragt, L.**,
Nielsen, D., Nielsen, D., Noga, K., O’Leary, A., Olson, R., Orr, A.,
Overgard Therkildsen, N., Palamartchouk, K., Perry, A., Peterson, M.,
Pipitone, J., Poisot, T., Pourreza, H., Povall, T., Richie-Halford, A.,
Ritchie, S., Rocha, C. B., Ross, N., Rydbeck, H., Sadjadi, M., Sanders,
S., Schloss, P., Seyffert, B., Shattow, G., Silva, R., Simpkin, S.,
Simpson, J., Smith, B., Sobhani, A., Soranzo, N., Srinath, A., Standage,
D., Staton, M., Steinbach, P., Stimberg, M., Telenczuk, B., Thoele, F.,
Timbers, T., Turner, S., van der Walt, A., van Schyndel, J., Vollmer,
D., von der Linden, J., Walker, A., Waterfall, J., Watson, G., White,
E., Williamson, E. P., Willing, C., Wilson, G., Winston, D., Wolff, H.,
Young, L., & Zamparo, L. (2017). Software Carpentry: The Unix Shell.
*Zenodo*. <https://doi.org/10.5281/zenodo.278226>



Allen, J., Bachant, P., Banaszkiewicz, P., Bekolay, T., Blischak, J.,
Boissonneault, M., Boughton, A., Cabunoc Mayes, A., Capes, G., Devenyi,
G. A., Digges, D., Duckles, J., Emonet, R., Fraser, A., Hansen, M.,
Hertweck, K., Irber, L., Jackson, M., King, W. T., Liu, G., Michonneau,
F., Mills, B., **Nederbragt, L.**, O’Leary, A., Pipitone, J., Poisot, T.,
Richie-Halford, A., Sherman, J., Silva, R., Smith, B., St-Onge, P.-L.,
Teucher, A., Williamson, E. P., & Wilson, G. (2017). Software Carpentry:
Automation and Make. *Zenodo*. <https://doi.org/10.5281/zenodo.278220>



Wilson, G., Becker, E., McKay, S., Michonneau, F., Williams, J. J.,
Mayes, A. C., Silva, R., (hoytpr), P., Crosby, K., **Nederbragt, L.**,
NielInfante, Psomopoulos, F. E., Gourlé, H., (gaiusjaugustus), G.,
dbmarchant, Emonet, R., Cock, P., Banaszkiewicz, P., Teal, T., HLindsay,
Thomas, A., Hamm, C., Allen, J., Jordan, K. L., EvanWill, Belkin, M.,
Pipitone, J., Young, N., Tang, M., Paudel, D., synesthesiam, Mills, B.,
Devenyi, G. A., Carroll, I., King, T. W., & gtehennepe. (2017). Data
Carpentry Wrangling Genomics Lesson. *Zenodo*.
<https://doi.org/10.5281/zenodo.1064254>
