---
layout: page
permalink: /cv
title: CV
---

Date: June 2018

Summary
---------

I am a Senior Lecturer ('Førstelektor') at the Department of Biosciences, University of Oslo. I am also head of education and training for the Centre for Bioinformatics, Dept. of Informatics, Univ. of Oslo.

At the Dept. of Biosciences, I lead the implementation of Computing in Science Education in the bachelor degree. As part of that I organise the first-semester course "Introduction to Computational Modelling for the Biosciences" (BIOS1100), teaching new Bioscience students mathematical modeling of biological problems using Python.

For the Centre for Bioinformatics, I develop and coordinate a portfolio of credited courses as well as workshops for master and PhD students and other staff.

I am a self-taught bioinformatician with a firm background in biology. My research at the Centre for Ecological and Evolutionary Synthesis ([CEES](https://www.mn.uio.no/cees/english/)) at the Dept. of Biosciences is on new methods to represent (reference) genomic information. I previously had a central role in the sequencing and assembly of the Atlantic cod genome. I lead the Advisory Council for eInfrastructure ([fagrådet for eInfrastruktur](www.uio.no/for-ansatte/nettverk-moter/los-enhetene/forskningsinfrastruktur/einfrastruktur/)) at UiO.

I am a Software Carpentry instructor, teaching computational lab skills to scientists in the Nordic countries. I am an advocate of open source software, and reproducible computational biology. I share my work not only through publications, but also on my science blog at lexnederbragt.com. My presentations can be found on [figshare](figshare.com/collections/Slides_of_my_talks_presentations/3488832) and [slideshare](http://slideshare.com/flxlex), and I am active on twitter ([@lexnederbragt](https://twitter.com/lexnederbragt)).

I have an analytical mind that loves problem solving. I value honesty and professionalism, and dislike hype. I believe that sharing and openness will further science more than competition. I choose impact over impact factor.


Research and Professional Experience
------------------------------------
* 2018- Senior lecturer, Dept. of Biosciences and Dept of Informatics, University of Oslo, Norway  
* 2018- Head of education and training, Centre for Bioinformatics, Dept of Informatics, University of Oslo, Norway  
* 2013-2018 Associate Professor (20%), Dept of Informatics, University of Oslo, Norway  
* 2005-2018  Postdoct. Res. Fellow, Researcher, Senior Engineer, Dept. of Biosciences, Univ. of Oslo, Norway  
* 2003-2005 Senior Scientist, Lingvitae AS, Oslo, Norway  
* 2002-2003 Postdoctoral research project University of Hawaii, USA  

Education
---------
* 2014-2015 Foundation programme Teaching and Learning in Higher Education ('Pedagogisk basiskompetanse'): Common part ('fellesdelen', 50 hours) + 37.5 hours in small modules  
* 1997-2001 PhD Biology, Utrecht Grad. School of Developmental Biology, Utrecht University, Netherlands  
* 1990-1996 BSc & MSc Biology, Utrecht University, Netherlands  


Teaching and teaching material
--------

**Courses organised**

BIOS1100 - Introduction to Computational Modelling in the Biosciences: <http://www.uio.no/studier/emner/matnat/ibv/BIOS1100>; new first-year bachelor course that I am coordinator and main teacher for; the course is the first implementation of the 'Computing in Science Education' project for the Biosciences study program, teaching python programming and modelling of real biological phenomena, 2017-

INF-BIO5121/9121 - High Throughput Sequencing technologies and bioinformatics analysis Univ. of Oslo: <http://inf-biox121.readthedocs.org/en/2015/>; I organise the course, coordinating with the five other teachers, and teach the assembly module: <http://inf-biox121.readthedocs.org/en/2016/Assembly/>, 2012-2016

**Contributions to university courses**

MBV-INF4410/9410 Bioinformatics for Molecular Biology (Univ. of Oslo): lecture on "The bioinformatics of sequencing and assembling genomes", and "What does it mean to do bioinformatics?", 2013-2016

BIO9905MERG1 - Bioinformatics for Metagenomic Analyses and Environmental Sequencing (Univ. of Oslo): lectures on "Next Generation Sequencing techniques and data relevant for metagenomics analyses" and "Assembly of metagenomes", 2011

BIO2120 Evolusjonsbiologi (Univ. of Oslo): lectures on "Evolution and Development" and "Evolution of Genes and Genomes"; group work, 2006 - 2007

Erasmus ICP course Marine Cell Biology (Observatoire Oceanologique, Banyuls-sur-mer, France): lectures on "Fundamental aspects of development" and "Cell cycle changes during development", 2000

**Contributions to workshops**

Next-Gen Sequence Analysis Workshop 'week 3' (intermediate and advanced skills) (invited) 2015, Michigan State University <http://angus.readthedocs.org/en/2015/week3.html>
Univ. California Davis Assembly Masterclass (invited) 2013
<http://davis-assembly-masterclass-2013.readthedocs.org/en/latest/>
*De novo* genome assembly, Univ. of Oslo, 2011
*De novo* genome assembly (invited), Univ. of Gothenburg, 2011

### Software and Data Carpentry
[Software Carpentry](http://software-carpentry.org/) and [Data Carpentry](http://www.datacarpentry.org/) are teaching researchers in science, engineering, medicine, and related disciplines worldwide computing and data analysis skills through short workshops using volunteer instructors.  I am a certified instructor of and organize and lead workshops in Norway and Sweden. Together with Karin Lagesen and with support from the Science Library (UiO), we have established Software and Data Carpentry at the University of Oslo, see [uio.no/carpentry](http://uio.no/carpentry), organising workshops more and more frequently, and increasing the number of certified instructors. I am also a certified *instructor trainer*, which enables me to give workshops for researchers that want to become instructors for Software or Data Carpentry.

**Software Carpentry workshops taught**
University of Oslo: 2012, 2013, 2015, 2016, 2017
Netherlands eScience Centre: 2017
University of Bergen: 2014
Science for Life Laboratory, Stockholm, Sweden 2014

**Software/Data Carpentry instructor retreat**
I gave a session on Interactive learning techniques, recording: <https://www.youtube.com/watch?v=QptHJgzooU0>.

Grants that I am listed on as collaborator
--------------------

### European Research council
ERC Consolidator Grant: **CICHLIDX**, an integrative approach towards the understanding of an adaptive radiation of East African cichlid fishes (PI: Walter Salzburger, Univ. of Basel)

### Research Council of Norway
**REPEAT: Evolutionary and functional importance of simple repeats in
the genome** 'Toppforsk' project (PI: Kjetill Jakobsen)
**Aqua Genome Project** (PI: Kjetill Jakobsen)
**Norwegian Sequencing Centre** Research Infrastructure, phase II (PI: Kjetill Jakobsen, Univ. of Oslo)
**CodReseq**: Translating the cod genome for aquaculture (PI: Kjetill Jakobsen)
**Haddoc**, assessment of long-term effects of oil exposure on early life stages of Atlantic haddock (PI: Sonic Meier, Inst. of Marine Research)
**Genome dynamics in early eukaryotic evolution**: importance of enigmatic lineages (PI: Russel Orr, Univ. of Oslo)
**The genomic and physiological basis of invasiveness in a harmful house-invader** (InHouse) (PI: Inger Skrede, Univ. of Oslo)

Selected blog posts
-------------------
"A new 1st semester bachelor course 'Introduction to Computational Modelling for the Biosciences'" <https://flxlexblog.wordpress.com/2017/03/08/a-new-1st-semester-bachelor-course-introduction-to-computational-modelling-for-the-biosciences>

"Developments in high throughput sequencing – July 2016 edition" <https://flxlexblog.wordpress.com/2016/07/08/developments-in-high-throughput-sequencing-july-2016-edition/>

"Active learning strategies for bioinformatics teaching" (2015)
<https://flxlexblog.wordpress.com/2015/08/31/active-learning-strategies-for-bioinformatics-teaching-2/>

"On graph-based representations of a (set of) genomes" (2015) <https://flxlexblog.wordpress.com/2015/04/09/on-graph-based-representations-of-a-set-of-genomes/>

"On the benefits of ‘open’ for teaching" (2014)
<https://flxlexblog.wordpress.com/2014/11/04/on-the-benefits-of-open-for-teaching/>


Software
--------

Github: <https://github.com/lexnederbragt>

*Selected repositories:*
RepSeq (Nederbragt et al 2010) <https://github.com/lexnederbragt/RepSeq>
SequenceTools (scripts to work with DNA sequence and related files) <https://github.com/lexnederbragt/sequencetools>

Invited Seminars
----------------
Slide decks:

* <http://www.figshare.com/collections/Slides_of_my_talks_presentations/3488832>
* <http://www.slideshare.net/flxlex>

Selected presentations:

"Genomic variation detection in Atlantic cod using long reads", SMRT Scientific Symposium, Leiden, The Netherlands, 2016 <https://figshare.com/articles/Genomic_variation_detection_in_Atlantic_cod_using_long_reads/3413887>

"Local, central or cloud: why we chose to build our e-infrastructure for sequence analysis embedded in the university of Oslo HPC cluster" workshop "e-infrastructures for Massively Parallel Sequencing", Uppsala, 2015 <https://www.youtube.com/watch?v=g5JtEQPEQiI>

"Coding & Best Practice in Programming: Why it matters so much in the NGS era", SeqAhead Scientific Meeting 2014 "NGS Data after the Gold Rush", Norwich, 2014 <http://www.slideshare.net/flxlex/140507-seq-aheadngscodingbestpractices>

"Experiences with Combining PacBio with Short Read Technology for Improved de novo Assembly of Complex Genomes", 4th Next Generation Sequencing Congress, London 2012 <http://www.slideshare.net/flxlex/131014-cees-holmenreproducibility>

Professional and Community Service
----------------------------------
**Grant Panels and Reviewing**
The Resource Allocation Committee for CPU and storage access at National HPC facilities (UNINETT Sigma2 AS), Norway (2014-)

**Advisory boards**
Secretary, Executive Council 'The Carpentries' (Software and Data Carpentry), 2018-
Leader, Advisory Council ('fagråd') eInfrastructure, University of Oslo (2016-)
Leader of the board, Digital Life Norway Research School (2016-)
MicrobesNG Microbial Genome Sequencing service and Open Strain Archive, Univ. of Birmingham, External Advisory Board (2014-)

**Manuscript Reviewing**
Open reviews: Publons <https://publons.com/author/18712/alexander-j-nederbragt>

Reviewer for Nature Methods, Nature Scientific Data, Bioinformatics, BMC Bioinformatics, BMC Genomics, Animal Genetics, Gigascience

**Volunteer activies**
Leader, 'kodeklubben Bærum' (code club), offering coding classes to children from 8 to 14 (2016-)
Board member of the residential condominium 'Hiltonåsen', Slependen, Norway 2009-2015

Research Mentoring
------------------
**Graduate Students**
Co-supervisor PhD students, Univ. of Oslo since 2008, 2 finished, 4 ongoing

**Undergraduate Researchers**
Main supervisor MSc students, Univ. of Oslo since 2014, 4 finished, 1 ongoing
Co-supervisor MSc students, Univ. of Oslo since 2013, 2 finished, 1 ongoing

Languages
---------

Dutch: first language, excellent oral/written  
English: excellent oral/written  
Norwegian: good oral/written  
German: conversational

List of publications
--------------------

## Preprints

Tørresen, O. K., Brieuc, M. S. O., Solbakken, M. H., Sørhus, E., **Nederbragt, A. J.**, Jakobsen, K. S., Meier, S., Edvardsen, R. B., & Jentoft, S. (2017). Genomic architecture of codfishes featured by expansions of innate immune genes and short tandem repeats. <em>BioRxiv</em>, 163949. <a href="https://doi.org/10.1101/163949" class="uri">https://doi.org/10.1101/163949</a>



Varadharajan, S., Sandve, S. R., Tørresen, O. K., Lien, S., Vollestad, L. A., Jentoft, S., **Nederbragt, A. J.**, & Jakobsen, K. S. (2017). The grayling genome reveals selection on gene expression regulation after whole genome duplication. <em>BioRxiv</em>, 153270. <a href="https://doi.org/10.1101/153270" class="uri">https://doi.org/10.1101/153270</a>



## Peer-reviewed publications

Elgvin, T. O., Trier, C. N., Tørresen, O. K., Hagen, I. J., Lien, S., **Nederbragt, A. J.**, Ravinet, M., Jensen, H., & Sætre, G.-P. (2017). The genomic mosaicism of hybrid speciation. <em>Science Advances</em>, <em>3</em>(6), e1602996. <a href="https://doi.org/10.1126/sciadv.1602996" class="uri">https://doi.org/10.1126/sciadv.1602996</a>



Khelik, K., Lagesen, K., Sandve, G. K., Rognes, T., & **Nederbragt, A. J.** (2017). NucDiff: In-depth characterization and annotation of differences between two sets of DNA sequences. <em>BMC Bioinformatics</em>, <em>18</em>, 338. <a href="https://doi.org/10.1186/s12859-017-1748-z" class="uri">https://doi.org/10.1186/s12859-017-1748-z</a>



Rand, K. D., Grytten, I., **Nederbragt, A. J.**, Storvik, G. O., Glad, I. K., & Sandve, G. K. (2017). Coordinates and intervals in graph-based reference genomes. <em>BMC Bioinformatics</em>, <em>18</em>, 263. <a href="https://doi.org/10.1186/s12859-017-1678-9" class="uri">https://doi.org/10.1186/s12859-017-1678-9</a>



Simovski, B., Vodak, D., Gundersen, S., Domanska, D., Abdulrahman Azab, Holden, L., Holden, M., Grytten, I., Rand, K., Drablos, F., Johansen, M., Mora, A., Lund-Andersen, C., Fromm, B., Eskeland, R., Gabrielsen, O. S., Ferkingstad, E., Sigve Nakken, Bengtsen, M., **Nederbragt, A. J.**, Thorarensen, H. S., Akse, J. A., Glad, I., Hovig, E., & Sandve, G. K. (2017a). Gsuite HyperBrowser version2.0b. GigaScience Database.



Simovski, B., Vodák, D., Gundersen, S., Domanska, D., Azab, A., Holden, L., Holden, M., Grytten, I., Rand, K., Drabløs, F., Johansen, M., Mora, A., Lund-Andersen, C., Fromm, B., Eskeland, R., Gabrielsen, O. S., Ferkingstad, E., Nakken, S., Bengtsen, M., **Nederbragt, A. J.**, Sif Thorarensen, H., Andreas Akse, J., Glad, I., Hovig, E., & Sandve, G. K. (2017b). GSuite HyperBrowser: Integrative analysis of dataset collections across the genome and epigenome. <em>GigaScience</em>. <a href="https://doi.org/10.1093/gigascience/gix032" class="uri">https://doi.org/10.1093/gigascience/gix032</a>



Wilson, G., Bryan, J., Cranston, K., Kitzes, J., **Nederbragt, L.**, & Teal, T. K. (2017). Good enough practices in scientific computing. <em>PLOS Computational Biology</em>, <em>13</em>(6), e1005510. <a href="https://doi.org/10.1371/journal.pcbi.1005510" class="uri">https://doi.org/10.1371/journal.pcbi.1005510</a>



Malmstrøm, M., Matschiner, M., Tørresen, O. K., Star, B., Snipen, L. G., Hansen, T. F., Baalsrud, H. T., **Nederbragt, A. J.**, Hanel, R., Salzburger, W., Stenseth, N. C., Jakobsen, K. S., & Jentoft, S. (2016). Evolution of the immune system influences speciation rates in teleost fishes. <em>Nature Genetics</em>, <em>48</em>(10), 1204–1210. <a href="https://doi.org/10.1038/ng.3645" class="uri">https://doi.org/10.1038/ng.3645</a>



Star, B., **Nederbragt, A. J.**, Jentoft, S., Grimholt, U., Malmstrøm, M., Gregers, T. F., Rounge, T. B., Paulsen, J., Solbakken, M. H., Sharma, A., Wetten, O. F., Lanzén, A., Winer, R., Knight, J., Vogel, J.-H., Aken, B., Andersen, Ø., Lagesen, K., Tooming-Klunderud, A., Edvardsen, R. B., Tina, K. G., Espelund, M., Nepal, C., Previti, C., Karlsen, B. O., Moum, T., Skage, M., Berg, P. R., Gjøen, T., Kuhl, H., Thorsen, J., Malde, K., Reinhardt, R., Du, L., Johansen, S. D., Searle, S., Lien, S., Nilsen, F., Jonassen, I., Omholt, S. W., Stenseth, N. C., & Jakobsen, K. S. (2011). The genome sequence of Atlantic cod reveals a unique immune system. <em>Nature</em>, <em>477</em>(7363), 207–210. <a href="https://doi.org/10.1038/nature10342" class="uri">https://doi.org/10.1038/nature10342</a>



**Nederbragt, A. J.**, Loon, A. E. van, & Dictus, W. J. A. G. (2002). Expression of Patella vulgata Orthologs of engrailed and dpp-BMP2/4 in Adjacent Domains during Molluscan Shell Development Suggests a Conserved Compartment Boundary Mechanism. <em>Developmental Biology</em>, <em>246</em>(2), 341–355. <a href="https://doi.org/10.1006/dbio.2002.0653" class="uri">https://doi.org/10.1006/dbio.2002.0653</a>



## Book chapters

Tørresen, O. K., Rise, M. L., Jin, X., Star, B., MacKenzie, S., Jakobsen, K. S., **Nederbragt, A. J.**, & Jentoft, S. (2016). 3 - An improved version of the Atlantic cod genome and advancements in functional genomics: Implications for the future of cod farming. In <em>Genomics in Aquaculture</em> (pp. 45–72). San Diego: Academic Press. Retrieved from <a href="http://www.sciencedirect.com/science/article/pii/B9780128014189000032" class="uri">http://www.sciencedirect.com/science/article/pii/B9780128014189000032</a>



Tørresen, O. K., Star, B., Jentoft, S., Jakobsen, K. S., & **Nederbragt, A. J.** (2016). 1 - The new era of genome sequencing using high-throughput sequencing technology: Generation of the first version of the Atlantic cod genome. In <em>Genomics in Aquaculture</em> (pp. 1–20). San Diego: Academic Press. Retrieved from <a href="http://www.sciencedirect.com/science/article/pii/B9780128014189000019" class="uri">http://www.sciencedirect.com/science/article/pii/B9780128014189000019</a>



## Software publications

Crusoe, M. R., Alameldin, H. F., Awad, S., Boucher, E., Caldwell, A., Cartwright, R., Charbonneau, A., Constantinides, B., Edvenson, G., Fay, S., Fenton, J., Fenzl, T., Fish, J., Garcia-Gutierrez, L., Garland, P., Gluck, J., González, I., Guermond, S., Guo, J., Gupta, A., Herr, J. R., Howe, A., Hyer, A., Härpfer, A., Irber, L., Kidd, R., Lin, D., Lippi, J., Mansour, T., McA’Nulty, P., McDonald, E., Mizzi, J., Murray, K. D., Nahum, J. R., Nanlohy, K., **Nederbragt, A. J**., Ortiz-Zuazaga, H., Ory, J., Pell, J., Pepe-Ranney, C., Russ, Z. N., Schwarz, E., Scott, C., Seaman, J., Sievert, S., Simpson, J., Skennerton, C. T., Spencer, J., Srinivasan, R., Standage, D., Stapleton, J. A., Steinman, S. R., Stein, J., Taylor, B., Trimble, W., Wiencko, H. L., Wright, M., Wyss, B., Zhang, Q., zyme, en, & Brown, C. T. (2015). The khmer software package: Enabling efficient nucleotide sequence analysis. <em>F1000Research</em>. <a href="https://doi.org/10.12688/f1000research.6924.1" class="uri">https://doi.org/10.12688/f1000research.6924.1</a>



## Published lesson materials

Achterberg, H., Adams, J., Adelman, J., Allen, J., Aranda, J., Bae, S., Banaszkiewicz, P., Barmby, P., Barr, E., Beitey, D., Bekolay, T., Berghold, J., Bjørnstad, M., Blischak, J., Boissonneault, M., Bosley, C., Bostroem, A., Boughton, A., Brase, R., Brown, A., Brown, K., Cabunoc Mayes, A., Carroll, I., Chase, J., Childs, S., Choi, J., Clare, R., Clayton, S., Cock, P., Connell, M., Corvellec, M., Coudrat, T., Dale, R., Davis, M., Davis, N., Davison, A., Demichelis, R., Devenyi, G. A., Dolson, E., Dotson, D., Duchesne, L., Duckles, J., Emonet, R., Endsley, K. A., Fauchereau, N., Ford, T., Förstner, K., Gonzalez, I., Gosmann, J., Gosset, J., Gray, J., Greshake, B., Guillou, S., Haley, M., Hames, S., Hamrick, J. B., Hannon, E., Hansen, M., Hertweck, K., Hinsen, K., Hjelm, J., Hodges, T., Howard, D., Irving, D., Jackson, M., Jolly, B., Jones, N., Joyce, B., Ketcheson, D., King, W. T., Klerke, S., Ko, L., Kwong, A., Laken, B., Lapp, H., Latornell, D., Leeman, J. R., Leyder, J.-C., Ligtenberg, W., Lin, J., Lonsdale, A., Manhaes Savio, A., May, R., Mazur, D., Michonneau, F., Mills, B., Mughal, Z., **Nederbragt, L.**, Neufeld, R., O’Leary, A., Obeng, A., Ory, J., Osiecka, N., Pipitone, J., Pińska, A., Poisot, T., Pomorski, P., Raghupathy, N., Rathgeber, F., Rawls, M., Richie-Halford, A., Riley, J., Robitaille, T., Rokem, A., Roswell, M., Sadjadi, M., Sales de Andrade, E., Schmeier, S., Sheneman, L., Shiffer, A., Shkurti, A., Silva, R., Soranzo, N., Spence, E., Srinath, A., Staneva, V., Stapleton, J., Stucky, B., Taylor, C., Teal, T., Telenczuk, B., Thomas, I., Thorne, B., Torres, G., Tröndle, T., Upani, J., Vahtras, O., Helm, E. van der, Walsh, A., Walter, N., Weitzel, D., Wheeler, D., White, E., Williamson, E. P., Willmore, F., Wilson, A., Wilson, G., Xiao, X., & Zonca, A. (2017, February). Software Carpentry: Programming with Python. Retrieved from <a href="https://zenodo.org/record/278222#.WUvbc3XytE4" class="uri">https://zenodo.org/record/278222\#.WUvbc3XytE4</a>



Ahmadia, A., Allen, J., Appling, A., Aubin, S., Bachant, P., Baird, D., Banaszkiewicz, P., Barmby, P., Batut, B., Bekolay, T., Belkin, M., Blischak, J., Bonsma, M., Borrelli, J., Boughton, A., Bouquin, D., Brauning, R., Brett, M., Brown, A., Cabunoc Mayes, A., Charlesworth, J., Charlton, B., Chen, D., Christensen, G., Collings, R., Corvellec, M., Davis, M., Devenyi, G. A., Dolson, E., Duchesne, L., Duckles, J., Emonet, R., Estève, L., Farsarakis, E., Fauber, B., Fouilloux, A., Förstner, K., Geiger, S., Gonzalez, I., Guarinello, M., Guillou, S., Hadwin, J., Haeni, M., Haessig, P., Hannah, N., Hansen, M., Harihareswara, S., Heilmaier, A., Heroux, M., Hertweck, K., Hilboll, A., Hinsen, K., Huang, D., Ismiraldi, Y., Jackson, M., Jacobs, C., Jarecka, D., Johnston, L. W., Jones, D., Jędrzejewski-Szmek, Z., Kelly, T., King, W. T., Kluyver, T., Konrad, B., Kuzak, M., Küderle, A., Labrie, K., Lapp, H., Latornell, D., Laufersweiler, M., LeBauer, D., Lee, K., Liffers, M., Llebot, C., Loucks, C., Ma, K., Marwaha, K., May, R., Michonneau, F., Mills, B., Mueller, A., Munk, M., Nagraj, V., **Nederbragt, L.**, Nunez-Iglesias, J., O’Brien, B., O’Leary, A., Olsson, C., Panitz, M., Pawsey, C., Pfenninger, S., Pipitone, J., Poisot, T., Preney, P., Rice, T., Riemer, K., Rio Deiros, D., Robinson, N., Rockenberger, A., Rohl, A., Rokem, A., Sacks, B., Sarahan, M., Schmeier, S., Schmider, H., Shellito, P., Shriwise, P. C., Silva, R., Smithyman, B., Soranzo, N., Steinbach, P., Stevens, S., Stueker, O., Stuermer, B., Timbers, T., Traphagen, D., Tröndle, T., Walt, A. van der, Vandervalk, S., Watson, G., Weaver, B., Wheelhouse, M., White, E., Williamson, E. P., Wilson, G., Wu, S., & Zhang, Q. (2017a, February). Software Carpentry: Version Control with Git. Retrieved from <a href="https://zenodo.org/record/278219#.WUvbgHXytE4" class="uri">https://zenodo.org/record/278219\#.WUvbgHXytE4</a>



Ahmadia, A., Allen, J., Banaszkiewicz, P., Becker, E., Bekolay, T., Belkin, M., Blischak, J., Boughton, A., Bray, E., Cabunoc Mayes, A., Carroll, I., Crouch, S., Davis, M., Davis, N., Devenyi, G. A., Duckles, J., Emonet, R., Fortin, F.-A., Gonzalez, I., Hamm, C., Hansen, M., Harris, R. M., Henninger, F., Hertweck, K., Hinsen, K., Hodge, A., Jackson, M., King, W. T., Kitzes, J., Koch, C., Lang, D., Liversidge, T., Michonneau, F., Mills, B., **Nederbragt, L.**, Nelson, E., O’Leary, A., Ortiz-Zuazaga, H., Patitsas, E., Pawlik, A., Perez, F., Pipitone, J., Poisot, T., Rokem, A., Silva, R., Steyn, J., Teal, T., Tröndle, T., Tweedie, F., Walt, A. van der, Vie, J.-J., Walker, J., Walsh, A., Weaver, B., White, E., Wilkerson, C., Williams, J., Williamson, E. P., Wilson, G., & Woo, K. (2017b, February). Software Carpentry: Instructor Training. Retrieved from <a href="https://zenodo.org/record/278229#.WUvbfHXytE4" class="uri">https://zenodo.org/record/278229\#.WUvbfHXytE4</a>



Aldazabal Mensa, I., Alexander, H., Allen, J., Alsheikh-Hussain, A., Attali, D., Baird, D., Baltzell, A., Banaszkiewicz, P., Barmby, P., Beagrie, R., Beane, G., Bedini, A., Bekolay, T., Belikov, E., Bell, J., Bill, B., Blin, K., Blischak, J., Boardman, S., Boissonneault, M., Bonnie, J., Boughton, A., Brase, R., Brown, A., Brunson, D., Bryk, J., Buske, O., Cabunoc Mayes, A., Carlise, M., Chen, D., Chung, K., Chung, K., Deveau, D., Devenyi, G. A., Doehle, P., Dolson, E., Doyle, M., Duckles, J., Emonet, R., Eyers, D., Fernandes, F., Fontenelle, H., Gacenga, F., Gardner, P., Gidden, M., Gonzalez, I., Gray, N., Guillou, S., Hagh, V. F., Hansen, M., Harstad, E., Hohenstein, T., Howe, A., Imamoglu, F., Irving, D., Jackson, M., Jane McTavish, E., Jennings, M., Jones, D., Jones, M. A., Keener, A., Keipert, K., Kelly, T., Kim, J. T., King, W. T., Koch, C., Konrad, B., Lake, S., Lang, D., Latornell, D., Lijnzaad, P., Ma, E., Mac, A., Madin, J., Magle, T., Mandel, M., Marini, C., Marwaha, K., Mashchenko, S., McCloy, D., Michonneau, F., Middleton, R., Milhans, J., Mills, B., Miotto, A., Mount, S., **Nederbragt, L.**, Nielsen, D., Nielsen, D., Noga, K., O’Leary, A., Olson, R., Orr, A., Overgard Therkildsen, N., Palamartchouk, K., Perry, A., Peterson, M., Pipitone, J., Poisot, T., Pourreza, H., Povall, T., Richie-Halford, A., Ritchie, S., Rocha, C. B., Ross, N., Rydbeck, H., Sadjadi, M., Sanders, S., Schloss, P., Seyffert, B., Shattow, G., Silva, R., Simpkin, S., Simpson, J., Smith, B., Sobhani, A., Soranzo, N., Srinath, A., Standage, D., Staton, M., Steinbach, P., Stimberg, M., Telenczuk, B., Thoele, F., Timbers, T., Turner, S., Walt, A. van der, Schyndel, J. van, Vollmer, D., Linden, J. von der, Walker, A., Waterfall, J., Watson, G., White, E., Williamson, E. P., Willing, C., Wilson, G., Winston, D., Wolff, H., Young, L., & Zamparo, L. (2017, February). Software Carpentry: The Unix Shell. Retrieved from <a href="https://zenodo.org/record/278226#.WUvbd3XytE4" class="uri">https://zenodo.org/record/278226\#.WUvbd3XytE4</a>



Allen, J., Bachant, P., Banaszkiewicz, P., Bekolay, T., Blischak, J., Boissonneault, M., Boughton, A., Cabunoc Mayes, A., Capes, G., Devenyi, G. A., Digges, D., Duckles, J., Emonet, R., Fraser, A., Hansen, M., Hertweck, K., Irber, L., Jackson, M., King, W. T., Liu, G., Michonneau, F., Mills, B., **Nederbragt, L.**, O’Leary, A., Pipitone, J., Poisot, T., Richie-Halford, A., Sherman, J., Silva, R., Smith, B., St-Onge, P.-L., Teucher, A., Williamson, E. P., & Wilson, G. (2017, February). Software Carpentry: Automation and Make. Retrieved from <a href="https://zenodo.org/record/278220#.WUvbbXXytE4" class="uri">https://zenodo.org/record/278220\#.WUvbbXXytE4</a>
