---
title: ""
---

## About me

I am a (Teaching) Professor ('Dosent') in Biology at the University of Oslo (UiO),
and have taught at the Faculty of Mathematics and Natural Sciences for many years.
I am an experienced educator whose main focus is students' learning.
I am interested in how we can increase educational competence
with those involved in my own and others' courses and teaching.

Reverse instructional design and constructive alignment
are two educational principles that are central to my work.
Student active learning is the starting point when I plan my teaching.
Integrating digital tools in a pedagogical way to promote student learning
is something that is central to everything I do.

I lead the implementation of "Computing in Science Education",
integration of computing and programming, in the Bioscience bachelor's programme.
As part of this, I have developed the first semester course
"Introduction to computational models for Biosciences" (BIOS1100/BIOS1101),
where I teach new bioscience students mathematical modelling
of biological problems using programming in Python.

My work at UiO resulted in me being appointed as 'Merittert Utdanner'
('distinguished educator') at the university in 2023. 
Read my application (in Norwegian) [here](https://lexnederbragt.github.io/soeknad_merittering).
In 2025 I was awarded the Olav Thon Foundation
'National award for outstanding teaching in the mathematical/natural sciences and medical fields'.

## Manuscripts I am most proud of


### Good enough practices

Wilson, G., Bryan, J., Cranston, K., Kitzes, J., Nederbragt, L., & Teal, T. K. (2017). Good enough practices in scientific computing. <em>PLOS Computational Biology</em>, <em>13</em>(6), e1005510. <a href="https://doi.org/10.1371/journal.pcbi.1005510" class="uri">https://doi.org/10.1371/journal.pcbi.1005510</a>

A true team effort, written by members of my favorite community, [the Carpentries](http://carpentries.org). A companion text to the ['Best practices'](https://journals.plos.org/plosbiology/article?id=10.1371/journal.pbio.1001745) paper, meant to present strategies that researchers can (and we feel, should) adopt, but also will keep using once they have adopted them.

### Live coding

Nederbragt, A., Harris, R. M., Hill, A. P., & Wilson, G. (2020). Ten quick tips for teaching with participatory live coding. <em>PLOS Computational Biology</em>, *16*(9), e1008090. <a href="https://doi.org/10.1371/journal.pcbi.1008090" class="uri">https://doi.org/10.1371/journal.pcbi.1008090</a>

I am a big fan of participatry live coding as a technique for teaching programming. This paper describes the technique and gives teaching tips for using it. It is another collaboration between members of the [Carpentries](http://carpentries.org) community, and I am proud to have initiated it.


### Patella engrailed and dpp

Nederbragt, A. J., Loon, A. E. van, & Dictus, W. J. A. G. (2002). Expression of Patella vulgata Orthologs of engrailed and dpp-BMP2/4 in Adjacent Domains during Molluscan Shell Development Suggests a Conserved Compartment Boundary Mechanism. <em>Developmental Biology</em>, <em>246</em>(2), 341–355. <a href="https://doi.org/10.1006/dbio.2002.0653" class="uri">https://doi.org/10.1006/dbio.2002.0653</a>

In my mind, the highlight paper from my PhD, which was in 'evo-devo', evolutionary developmental biology. We showed expression of Engrailed and Dpp orthologs in early limpet development. The cells expressing these genes were on opposite sides of the border of the developing shell. We coupled that observation to the expression in cells at other borders in developing embryos.

### Atlantic cod genome I

Star, B., Nederbragt, A. J., Jentoft, S., Grimholt, U., Malmstrøm, ..., Omholt, S. W., Stenseth, N. C., & Jakobsen, K. S. (2011). The genome sequence of Atlantic cod reveals a unique immune system. <em>Nature</em>, <em>477</em>(7363), 207–210. <a href="https://doi.org/10.1038/nature10342" class="uri">https://doi.org/10.1038/nature10342</a>

We were able to use the Roche/454 GS FLX in a tru team-based effort to sequence the genome of the Atlantic Cod, the eukaryote first genome sololey sequenced using 'next-generation' technologies to be accepted into the Ensembl database. The genome showed that Atlantic cod has an unusual immune system, the first genome to show lack of key genes that 'everyone' thought vertebrates could not do without.

### Atlantic cod genome II

Tørresen, O. K., Rise, M. L., Jin, X., Star, B., MacKenzie, S., Jakobsen, K. S., Nederbragt, A. J., & Jentoft, S. (2016). 3 - An improved version of the Atlantic cod genome and advancements in functional genomics: Implications for the future of cod farming. In <em>Genomics in Aquaculture</em> (pp. 45–72). San Diego: Academic Press. Retrieved from <a href="http://www.sciencedirect.com/science/article/pii/B9780128014189000032" class="uri">http://www.sciencedirect.com/science/article/pii/B9780128014189000032</a>

My first last-author publication. A lot of work, with all available seuqencing technologies (at the time), many bioinformatics analyses led to a very complete Atlantic cod genome. I am proud to have contributed to making the Atlantic Cod genome continue to be a valuable resource to the community.


### Lessons

I am also proud to have contributed to [several Carpentries lessons](https://zenodo.org/search?page=1&size=20&q=lex%20nederbragt&type=lesson).

LICENSE NOTICE. Unless prevented by existing licenses or otherwise, I
release the material on this blog under a [CC0 license](http://creativecommons.org/publicdomain/zero/1.0/). Feel free
to use the material for the greater good. The license does not require a reference to the source, but I would appreciate it.


## Blog

Read my blog [here](blog/index.qmd).

## About this website

Based on [Reproducible Research Oxford's website](https://github.com/RROxford/rroxford.github.io), which based it on [Beautiful Jekyll](https://github.com/daattali/beautiful-jekyll). Source can be found at this [GitHub repository](https://github.com/lexnederbragt/lexnederbragt.com).
