---
layout: post
title: On assembly uncertainty (inspired by the Assemblathon2 debate)
date: 2013-02-26 15:21:21.000000000 +01:00
type: post
published: true
status: publish
categories:
- Bioinformatics
tags:
- assembly
meta:
  _edit_last: '12058266'
  draftfeedback_requests: a:2:{s:23:"o.k.torresen@ibv.uio.no";a:3:{s:3:"key";s:13:"512ca057380f8";s:4:"time";s:10:"1361879127";s:7:"user_id";s:8:"12058266";}s:24:"bastiaan.star@ibv.uio.no";a:3:{s:3:"key";s:13:"512ca0574985a";s:4:"time";s:10:"1361879127";s:7:"user_id";s:8:"12058266";}}
  draft_feedback: "a:1:{s:23:\"o.k.torresen@ibv.uio.no\";a:1:{i:0;a:2:{s:4:\"time\";s:10:\"1361880179\";s:7:\"content\";s:630:\"
    Du kan eventuelt ta med UMDs valideringsverktøy også: http://sourceforge.net/p/assembly-eval/code/183/tree/
    (Sergey nevnte det i en epost).\n\nDu er vel inne på det, men ja, de forskjellige
    tools trenger ofte forskjellige input. FRCbam nevner ikke noe spesielt, CGAL vil
    ha absolutt alt mappa så mange ganger det kan (usikker på hvor lang tid det vil
    ta på store genomer), mens Hagfish vil sette et vidt vindu. \n\nOg så er det hvordan
    mate pairs skal mappes. For FRCbam må de være orientert mot hverandre, og det
    er kanskje anderledes for andre tools (husker ikke om noen har eksplisit sagt
    noe).\n\nKom ikke på noe annet.\n\n\";}}}"
  _publicize_pending: '1'
  _oembed_7102c7ab012e6fc97d9c672084498576: '{{unknown}}'
  _oembed_1652f59009b8df61bbd6dece7d57c346: '{{unknown}}'
  _oembed_4ad299bbf1784afe101ec32d3b789c8a: '{{unknown}}'
  _oembed_9a42091ec8560f46137406fddab471de: '{{unknown}}'
  _oembed_dc617e499232bf2029f7cb82b7fc8c88: '{{unknown}}'
  _oembed_b35c27cbab2bb460051f9a7678031f1b: '{{unknown}}'
  _oembed_318b45a658d42fb71138b65db10e38a6: '{{unknown}}'
  _oembed_0955117e43f3b057729da3e4c0124e6e: '{{unknown}}'
  _oembed_a4c0f730f16fff11ff04b09b3c3fe58e: '{{unknown}}'
  _oembed_ef881a9e2c6f343fa5ad9f7358670055: '{{unknown}}'
  _oembed_8d22183b2f9ac2c4aa68b6cb60d36a44: '{{unknown}}'
  _oembed_2864d1a19f4e9bbf3b7a8e8bb6d95d4c: '{{unknown}}'
  _oembed_77075abb48a68bf78064d8b10f69dc43: '{{unknown}}'
  _oembed_cd923cc03d9c64319cbccbba8b4ffa5c: '{{unknown}}'
  _oembed_9fcb2d29ba2bcf8959361bb219c68693: '{{unknown}}'
  _oembed_99ef75b72454430f75fffcfeb3ddbe46: '{{unknown}}'
  _oembed_c9b42a8ec1eb3e45f26fa31e2d40a4a2: '{{unknown}}'
  _oembed_aab3cc939b66eb89f392295284acc2fb: '{{unknown}}'
  _oembed_b5e3ebe21f72d08d08a2c93ac4e12085: '{{unknown}}'
  _oembed_ca90cfdaccb47379b863c44e27292f68: '{{unknown}}'
  _oembed_c0d104bf9a9177d44693095ed9953ed4: '{{unknown}}'
  _oembed_a8f714c427868dc992824521cda03a44: '{{unknown}}'
  _oembed_4f6c5bf064729468ee1ecb77ea2b2ba4: '{{unknown}}'
  _oembed_3eb8e3076070fc84018d828800aee2e1: '{{unknown}}'
  _oembed_b64cfb9a79830fa5c545383223cfd73a: '{{unknown}}'
  _oembed_0f1653fd49b7b60e40fc97d9a98dba48: '{{unknown}}'
  _oembed_e7d303ae42ef921901367b1023459731: '{{unknown}}'
  _oembed_ace142c2f663f1cee3d3f10457b34fbe: '{{unknown}}'
  _oembed_894e3dd110abc1b091e7029e7a77a9e3: '{{unknown}}'
  _oembed_d982f6e462b43c376da2b7f404e8e6f3: '{{unknown}}'
  _oembed_57087c9f52d371b3ba40223553b8aba6: '{{unknown}}'
  _oembed_b01082500ff1a24cb4152ab933f8a046: '{{unknown}}'
  _oembed_bc2c391290d5572fb2e704ef6a1a65c0: '{{unknown}}'
  _oembed_cfa5439e01341891ca8fca4ca5ce71f3: '{{unknown}}'
  _oembed_35b06c276464b1ffc8ee8d7d84315ee7: '{{unknown}}'
  _oembed_97217531c6b89843c8f61d50052bee9f: '{{unknown}}'
  _oembed_917f3afe066a9c0bdca03b06e5024155: '{{unknown}}'
  _oembed_7e263123e971a5c89a8ad17e7353a292: '{{unknown}}'
  _oembed_dcd0e609cbbc144df0f80cec87afbab7: '{{unknown}}'
  _oembed_ad0048f2b72dd014c61f19123a3055d8: '{{unknown}}'
  _oembed_8482a786904d932ab7c89286466e6cae: '{{unknown}}'
  _oembed_4a1a50232643b3a90ffca2f0454ebe3e: '{{unknown}}'
  _oembed_94578a41c70caaea7d34ef11031b403c: '{{unknown}}'
  _oembed_266c2d2d12c80006bdec94e09ee633b0: '{{unknown}}'
  _oembed_888def2b013b93d9d2bd961ad241904b: '{{unknown}}'
  _oembed_6aa9cf0f94a1a7a03b9d5850927b8990: '{{unknown}}'
  _oembed_eb0f695cd06888b435b95020d2d5418d: '{{unknown}}'
  _oembed_598ade141f0698419b3615ef996c658d: '{{unknown}}'
  _oembed_20b27b71b36cd63b359e5837127d7a6f: '{{unknown}}'
  _oembed_a35ce00f6d7c9753f3dccc605970840a: '{{unknown}}'
author:
  login: lexnederbragt
  email: lex.nederbragt@bio.uio.no
  display_name: lexnederbragt
  first_name: ''
  last_name: ''
---
<p style="text-align:left;">The release of <a href="http://arxiv.org/abs/1301.5406">the assemblathon2 paper via arXiv</a>, <a href="http://ivory.idyll.org/blog/thoughts-on-assemblathon-2.html">the blog post</a> by Titus Brown and his <a href="http://ivory.idyll.org/blog/2013-assemblathon-review-i.html">posting of his review</a> of the paper sparked a discussion in the comments section of Titus' blog, as well as <a href="http://www.homolog.us/blogs/2013/02/23/titus-browns-thoughts-on-the-assemblathon-2-paper/">on the  homolog_us blog</a>. With this blog post, I'd like to chip in with a few remarks of my own.</p>
<p style="text-align:left;">First, I agree with all of what Titus Brown said ('you know, what he said'). One of his main take-home messages from the Assemblathon2 is the uncertainty associated with any assembly, and how this needs to be communicated better. When I give presentations to introduce 'this thing called assembly', I often start out with a quote from the Miller <em>et al.</em> 2010 paper in Genomics (<a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2874646/">'Assembly algorithms for next-generation sequencing data'</a>):</p>
<blockquote>
<p style="text-align:left;">An assembly is a hierarchical data structure that maps the sequence data to a putative reconstruction of the target</p>
</blockquote>
<p style="text-align:left;"><!--more-->I stress this aspect of the assembly being a 'putative reconstruction' first. I then move on to explain the hierarchy from reads to contigs to scaffolds (and perhaps superscaffolds/pseudochromosomes). In the future, I intend stress the fact that an assembly is a hypothesis of the genome more than I already do. Like Titus Brown said:</p>
<blockquote>
<p style="text-align:left;">[...] each assembly is a computational hypothesis, developed from noisy data using approximate computation, and that this assembly must be treated with skepticism</p>
</blockquote>
<p style="text-align:left;">This brings me to my main point: I think we are more then ever positioned to quantify the uncertainty of an assembly, and report both globally and locally how trustworthy it is. The Assemblathon2 mostly used data not involved in building the assembly, such as fosmid sequenced and optical maps. Titus Brown rightly points out in his review that RNA-seq (transcriptomics) data could also be used. I would like to add another aspect: checking that the assembly you obtain is in fact in correspondence with the data you used to built it. Do the reads agree with the assembly? As Titus says, did all reads actually end up being used?</p>
<p style="text-align:left;">The following tools are <em>already</em> available for this:</p>
<ul>
<li>REAPR, used for the Assemblathon2 validations, is 'a tool that evaluates the accuracy of a genome assembly using mapped paired end reads, without the use of a reference genome for comparison'. <a href="http://www.sanger.ac.uk/resources/software/reapr/">Link</a></li>
<li>FRCBAM extends the <a href="http://www.plosone.org/article/info:doi/10.1371/journal.pone.0031002">FRC</a> approach, detecting and quantifying 'features' possibly indicating assembly problems, based on mapping reads back to the assembly. For their paper on FRCBAM, the authors actually tested their approach on the Assemblathon2 results. <a href="https://github.com/vezzi/FRC_align">Link</a>, <a href="http://arxiv.org/abs/1210.1095">preprint on arXiv</a></li>
<li>ALE ('Assembly Likelihood Evaluation') uses aligned reads and k-mer distributions to determine a likelihood score. ALE is meant as a comparative measure, i.e. to compare different assemblies by their ALE score. <a href="http://sc932.github.com/ALE/">Link</a>, <a href="http://bioinformatics.oxfordjournals.org/content/early/2013/01/09/bioinformatics.bts723.short">paper</a></li>
<li>CGAL is another likelihood based method.  <a href="http://bio.math.berkeley.edu/cgal/">Link</a>, <a href="http://genomebiology.com/2013/14/1/R8">paper</a></li>
</ul>
<p>Note that all these tools start by (having you) mapping back one or more of the input read datasets to the assembled sequences, and then using different methods/algorithms to score fidelity of the sequences, and/or detect possible misassemblies. Locally aligned read depth, insert sizes between paired reads, orientations of the pairs are all indicators used by these approaches. Most importantly, none of these tools need a reference sequence for the task!</p>
<p style="text-align:left;">Notably, many of these tools are able to generate information on assembly uncertainty along the assembled sequences, information which can be transformed into genomic tracks (e.g., through gff files) to be used in a genome browser. This would allow for supplying these tracks with a finished assembly.</p>
<p style="text-align:left;">As some of the readers of this blog may be aware, we are working on a new version of the Atlantic cod genome. We intend to try out these tools (we haven't gotten very far with this yet), and see if we can generate tracks in an appropriate format. Hopefully, we will be able to develop a way to integrate the information of all the tracks into a final one with a summary 'assembly (un)certainty' score along the assembled sequences. When this score drops below a certain threshold, we intend to break the assembly (REAPR has implemented this already) to improve the final assembly.</p>
<p style="text-align:left;">On important caveat of this approach is that the tools describe depend on mapping reads back to the assembly. This poses its own problems, as mapping, like assembly, is far from a solved problem. Nor can, for that matter, one mapping exercise serve all tools. Not only have we had two assemblathons, and other such projects (GAGE, dnGASP), now two or perhaps more alignment competitions are starting (<a href="http://compbio.soe.ucsc.edu/alignathon/">Alignathon</a> and <a href="http://bioinf.dimi.uniud.it/variathon">Variathon</a>). So I expect some fiddling with the alignments to get all the tools to work properly (not looking forward to that :-( ).</p>
<p style="text-align:left;">In conclusion, I think we are more than able to report on uncertainties - and obviously certainties - associated with the hypothesis we call the assembly. I think it is time this reporting, for example in the form of annotation tracks visualisable in a genome browser, becomes standard for new assemblies. It must be said, though, that whether people actually are going to <em>use</em> this information remains to be seen. How many programs take per-base quality values of the reference into account when mapping reads, or of gene sequences when generating phylogenies (not my field, so I may be very off here)?</p>
<p style="text-align:left;">By the way, for certain reference genomes (human, mouse and zebrafish), tracks with clone-derived assembly problems are already available, see <a href="http://genomeref.blogspot.no/2012/07/hidden-assembly-problems-exposed.html">this blog post</a> from <a href="http://www.ncbi.nlm.nih.gov/projects/genome/assembly/grc/">The Genome Reference Consortium</a>.</p>
<p style="text-align:left;">Let's hope that by using methods to explicitly report assembly uncertainties, we can make the end-users of our 'reference' assemblies more aware of the strengths and weaknesses of the material they work with.</p>
<p style="text-align:left;">(Note that I left out <a href="http://bioinf.spbau.ru/quast">QUAST</a> and <a href="http://sourceforge.net/apps/mediawiki/amos/index.php?title=Amosvalidate">Amosvalidate</a>+<a href="http://www.plosone.org/article/info:doi/10.1371/journal.pone.0031002">FRC</a>, the first because it is not very useful without a reference sequence, the others because these require a file provided by the assembler describing the read layout, and there are very few assemblers able to generate this file)</p>
