---
layout: post
title: A Software Carpentry-inspired workshop to improve the way we do bioinformatics
  in our group
date: 2013-10-21 13:37:01.000000000 +02:00
type: post
published: true
status: publish
categories:
- Bioinformatics
tags:
- reproducible science
- Software Carpentry
- Teaching
meta:
  _edit_last: '12058266'
  draftfeedback_requests: a:1:{s:13:"525e98420e969";a:3:{s:3:"key";s:13:"525e98420e969";s:4:"time";s:10:"1381931074";s:7:"user_id";s:8:"12058266";}}
  _publicize_pending: '1'
author:
  login: lexnederbragt
  email: lex.nederbragt@bio.uio.no
  display_name: lexnederbragt
  first_name: ''
  last_name: ''
---
<p>About a year ago, I attended a Software Carpentry Bootcamp. <a href="http://software-carpentry.org/">Software Carpentry</a> aims 'to make scientists more productive, and their work more reliable, by teaching them basic computing skills'. As I described in a <a href="/2013/09/23/software-carpentry-bootcamp-lessons-learned/">previous blogpost</a>, attending the bootcamp changed many aspects of the way I work. I also decided to become a Software Carpentry instructor. Together with Karin Lagesen, we recently taught a BootCamp in Oslo. Our group at the <a href="http://www.mn.uio.no/cees/english/">Centre for Ecological and Evolutionary Synthesis </a>(CEES) is working on different aspects related to fish genomics. This more or less started with our leading role in the project for the sequencing and assembly of the genome of Atlantic cod. As part of that project, many group members had to learn basic unix shell commands, making small pipelines, running programs, and sometimes a bit of scripting. Now, we see many more people at the CEES are moving into bioinformatics, often as a result of them starting to get high-throughput sequencing data. Thus, we have many self-taught bioinformaticians at the centre, in other words, the ideal target audience for Software Carpentry.</p>
<p>To help the bioinformatics work in our group apply the principles of Software Carpentry, we are going to have an 'extended' workshop, spread over several weeks with one half-day session each week. In this post, I will present the intended subjects. <!--more-->Each one will be introduced by a teacher (some by invited persons, some by me), followed by practical application of the material, preferably on people's own data. In between sessions, participants are suggested to do some exercises, and further try out the material so that it sinks in. I intend to report on our progress and experiences on this blog. For all modules, a prerequisite is comfortable knowledge of the unix shell. The exact order is still a bit up in the air. Feedback welcome (in the comments-section)!</p>
<ul>
<li>Effective use of the shell
<ul>
<li>basic shell scripting</li>
<li>for loops</li>
<li>how to automate and abstract when possible and practicle</li>
</ul>
</li>
<li>Version control I: basics
<ul>
<li>basics: setting up a repo, committing, diff, log, turning back the clock</li>
<li>setting up and using a personal (university) repository</li>
<li>using a collaborative repository</li>
<li>working with public repositories</li>
</ul>
</li>
<li>Using 'make' to automate your work and make it more reproducible
<ul>
<li>basics of make</li>
<li>executing and developing your pipeline through a makefile</li>
<li>making your work reproducible with make</li>
</ul>
</li>
<li>From complex shell command pipeline to python script
<ul>
<li>no python knowledge is necessary, but it surely will help. At the very least, this module will hopefully inspire you to learn (more) python</li>
<li>examples of unix pipelines from participants</li>
<li>rework(ed) into small python script and explained</li>
</ul>
</li>
<li>IPython notebook
<ul>
<li>basic usage</li>
<li>reporting work in</li>
<li>sharing notebooks</li>
</ul>
</li>
<li>How we organise the bin, src, scripts folders
<ul>
<li>cleanup and reorganisation of the bin folder</li>
<li>differentiate source code files, binaries and scripts</li>
<li>using the module system</li>
</ul>
</li>
<li>Porting your analysis pipeline over to Galaxy
<ul>
<li>basics of Galaxy</li>
<li>making a workflow page to be distributed with your paper</li>
</ul>
</li>
<li>Reproducible research
<ul>
<li>practical aspects of enhancing reproducible research</li>
<li>tools to help achieve reproducible research</li>
<li>sharing of code, workflows, data</li>
</ul>
</li>
<li>Version control II: advanced and collaborative code development
<ul>
<li>branching and merging</li>
<li>pull requests</li>
</ul>
</li>
</ul>
