---
layout: post
title: My review of "Automated ensemble assembly and validation of microbial genomes"
date: 2014-06-05 12:40:38.000000000 +02:00
type: post
published: true
status: publish
categories:
- Bioinformatics
- Reviews
tags:
- assembly
meta:
  _wpcom_is_markdown: '1'
  _edit_last: '12058266'
  geo_public: '0'
  publicize_google_plus_url: https://plus.google.com/113347535998207715225/posts/VvP8viJSo3k
  _wpas_done_6015510: '1'
  _publicize_done_external: a:1:{s:11:"google_plus";a:1:{s:21:"113347535998207715225";b:1;}}
  _wpas_skip_6015510: '1'
  _oembed_2b546475bd238a26320e85fce9bd1a4f: '{{unknown}}'
  _oembed_7d7e21161ac7a4e7e87ed7361844b778: '{{unknown}}'
  _oembed_875d27e2dcf033853651be777700b0b2: '{{unknown}}'
  _oembed_0d95aa5290bfeaf27730e6afbacedd5f: '{{unknown}}'
  _oembed_cf0bd9da3d27c276c2f3db8f7d3da9e5: '{{unknown}}'
  _oembed_e8a27088b89ccd0a14065c6b88966cd3: '{{unknown}}'
  _oembed_06e54faec46f81bb1a6fa4cf46fef145: '{{unknown}}'
  _oembed_bd9986b02d79fa1137aad067b24d4629: '{{unknown}}'
  _oembed_8c92b98eee080f622b5984f88cf831c3: '{{unknown}}'
  _oembed_0d2260252631224d5e57c8744f2aabf6: '{{unknown}}'
  _oembed_2910e544a1b7ac8d2ea7f26d220a6fab: '{{unknown}}'
  _oembed_364387a075984203986422893132f2dd: '{{unknown}}'
  _oembed_22634a12dad3a22d00ab97f86b5c9402: '{{unknown}}'
  _oembed_71466816fc4ce35d20c20de0666381a5: '{{unknown}}'
  _oembed_53f8503fd49161c3a77eec180f4ab59e: '{{unknown}}'
author:
  login: lexnederbragt
  email: lex.nederbragt@bio.uio.no
  display_name: lexnederbragt
  first_name: ''
  last_name: ''
---
<p>Last month, a new paper appeared in <a href="http://www.biomedcentral.com/1471-2105/15/126/abstract">BMC Bioinformatics</a>, entitled "Automated ensemble assembly and validation of microbial genomes". In it, the authors describe <a href="http://www.cbcb.umd.edu/software/imetamos">iMetAMOS</a>, a module of the metAMOS package, for bacterial genome assembly. I was one of the reviewers (I signed my review), and post part of my review here. The full review can be found on <a href="https://publons.com/review/3544/">publons</a>.</p>
<p>[caption id="attachment_556" align="alignright" width="413"]<a href="http://www.biomedcentral.com/1471-2105/15/126/figure/F1"><img class="wp-image-556 size-large" src="{{ site.baseurl }}/assets/imetamos_workflow.jpg?w=413" alt="iMetAmos workflow. From the paper, doi:10.1186/1471-2105-15-126" width="413" height="1024" /></a> iMetAmos workflow. From the paper, doi:10.1186/1471-2105-15-126[/caption]</p>
<p>I signed my review because I believe in non-anonymous peer review (see Mick Watson's <a href="http://biomickwatson.wordpress.com/2013/02/11/the-reviewers-oath/">"reviewer’s oath"</a>).</p>
<p>I made my review available on <a href="http://publons.com">publons</a>, a platform to post pre- and post-publication peer-review reports after the article has been published, because I believe in open peer-review. EDIT Adam Phillippy, the senior author on the paper, posted the authors response to the review reports they received as a comment to review on publons!</p>
<p>I post the first part of my review here because it nicely summarises the paper and my (favourable) opinion of it. I'll admit that I wrote these paragraphs of the review report with the idea of posting them to my blog :-)</p>
<p>Enjoy!</p>
<hr />
<p><!--more-->The field of genome assembly has seen some interesting development recently. First, we have seen several assembly competitions (Assemblathon 1&amp;2, GAGE and GAGE-B) often using genomes with an available reference, where no real winners could be called. Second, new applications became available for reference-free evaluation of assemblies, based on the reads that were used for generating the assembled contigs and scaffolds. Conclusions I drew from these efforts, and our own work, are that each assembly project is unique and should be performed as a mini-assemblathon, followed by carefully selecting the best assembly for the goal of the project. Or, in other words, running multiple assemblers and using reference-free validation tools to judge them. It was therefore with great pleasure that I saw the pre-print for the iMetAMOS paper, and also the reason I happily agreed to review the manuscript.</p>
<p>iMetAMOS aims to combine these two steps of assembly using multiple programs, with a validation by employing again multiple tools, and then choses an assembly based on these evaluations. In addition, iMetAMOS performs a standard contamination check - or rather, is treats each assembly as a metagenomic one, determining the possible phylogenetic origin of all contigs/scaffolds. This makes it a unique tool, on that has not been available until now. I wholeheartedly support this effort.</p>
<p>The manuscript was a pleasure to read, easily understandable and short. I have mostly small comments, see below. iMetAMOS incorporates a fantastic suite of current state-of-the-art open source tools, and should perform well straight out of the box. I didn't consider anything missing that would be essential for its goals. Well, may be one. Perhaps a future version could add an assembly polishing/improvement tools such as Pilon (from the Broad Institute, <a href="http://www.broadinstitute.org/software/pilon/">http://www.broadinstitute.org/software/pilon/</a>).</p>
<p>In addition, I appreciate the possibility that user can add their own tool if they want (lack of time prohibited me from trying this out). I was able to successfully download, install and use the single file binary, albeit with some hiccups (see below). The quick test using a lambda phage dataset finished, as did the running of the ERR234097 example mentioned on <a href="http://www.cbcb.umd.edu/software/imetamos">http://www.cbcb.umd.edu/software/imetamos</a>. However, for ERR234097, I did not obtain the results mentioned in the supplementary excel file, see below.</p>
<p>An off-the-shelf tool like iMetAMOS is great for push-button bioinformatics. However, it allows the user to get away with not fully understanding the tools, parameters etc of the workflow. Here, we have to trust the authors to have done a thorough job, reporting on the choices made, and enabling thorough evaluation of the underlying material. First, the authors have a long track record in the field and I fully trust they know what they are doing. Second, all intermediate results are saved in the project directory, and all steps are logged in extensive log files. So, an interested user can (and, I would say, probably should) go and inspect each step of the procedure. If anything, there is an overload of information, making finding details a bit hard.</p>
<p>In general, the authors seem to aim to make life easier for scientists with little bioinformatics experience. For example, making available a single binary, with all dependencies included, takes the burden away from having to download and install a lot of software. The program generates a single html page at the end as point-of-entry for evaluating the results, and downloading the assembly. All these aspects I applaud. However, there are a few improvements possible here. Mainly, the documentation could be improved. The 'readthedocs' pages (<a href="http://metamos.readthedocs.org/en/latest/index.html">http://metamos.readthedocs.org/en/latest/index.html</a>) have very little information on how to run the software. Installation instructions assume one knows how to extract a tar.gz file. The github pages <a href="https://github.com/marbl/metAMOS">https://github.com/marbl/metAMOS</a> are more detailed, but, as an example, lack a description of what the different steps of the workflow do (Preprocess, Assemble, FindORFs, …). At the end of running the software, unless you know where to look, it is not straightforward to find the html pages generated. So in general, the software could be made easier to use for a beginner. (there are also three entry points, readthedocs, github and the http://www.cbcb.umd.edu/ pages) which I found confusing.</p>
<p>In conclusion, I would recommend iMetAMOS to users, both those with little bioinformatics experience, as well as super users, wanting to perform bacterial genome assemblies using short-read sequencing data.</p>
<hr />
<p>For the complete review report, see <a href="https://publons.com/review/3544/">https://publons.com/review/3544/</a>.</p>
