---
layout: post
title: Developments in high throughput sequencing – June 2015 edition
date: 2015-06-17 10:47:33.000000000 +02:00
type: post
published: true
status: publish
categories:
- Next Generation Sequencing
tags:
- '454'
- GS FLX+
- GS Junior
- HiSeq
- Illumina
- Ion Proton
- Ion Torrent
- Ion Torrent PGM
- MinIon
- MiSeq
- NextSeq500
- oxford nanopore
- pacbio
- sanger
- SOLiD
meta:
  _wpcom_is_markdown: '1'
  _edit_last: '12058266'
  geo_public: '0'
  _publicize_job_id: '11754970804'
  publicize_google_plus_url: https://plus.google.com/113347535998207715225/posts/5FMvEJXTqqp
  _publicize_done_6048082: '1'
  _wpas_done_6015510: '1'
  _wpas_skip_6015510: '1'
  _oembed_6e5ae8fbb805716721119bfb331bfb3b: '{{unknown}}'
  _oembed_8cf569ee8d3938ecd0359f8bfe006aea: '{{unknown}}'
  _oembed_6584ea7d6cd47af59f486d8c871f3ea3: '{{unknown}}'
  _oembed_5029987f44b952ca9cdb920f4c045153: '{{unknown}}'
  _oembed_ca58ee6d0fd5f01fae9e467896d36bce: '{{unknown}}'
  _oembed_0765a39d5fe58109f8860d4a34cd8420: '{{unknown}}'
  _oembed_1b3ce60fdf084c04fa7615970e9a408c: '{{unknown}}'
  _oembed_7fa366461745b18d212bdd115c888a19: '{{unknown}}'
  _oembed_31d3a677e692eab3b370a169b5236645: '{{unknown}}'
  _oembed_7ba995da6250c1dbc409cd01cd02beca: '{{unknown}}'
author:
  login: lexnederbragt
  email: lex.nederbragt@bio.uio.no
  display_name: lexnederbragt
  first_name: ''
  last_name: ''
---
<p>This is the fourth edition of this visualisation, previous editions were in <a href="/2014/06/11/developments-in-next-generation-sequencing-june-2014-edition/">June 2014</a>, <a href="/2013/10/01/developments-in-next-generation-sequencing-october-2013-edition/">October 2013</a> and <a href="/2012/12/03/developments-in-next-generation-sequencing-a-visualisation/">December 2012</a>.</p>
<p>As before, full run throughput in gigabases (billion bases) is plotted against single-end read length for the different sequencing platforms, both on a log scale. Yes, I know a certain new instrument seems to be missing, hang on, I'm coming back to that...</p>
<p><!--more--></p>
<p><a href="https://flxlexblog.files.wordpress.com/2015/06/developments_in_high_throughput_sequencing.jpg"><img class="alignnone size-large wp-image-678" src="{{ site.baseurl }}/assets/developments_in_high_throughput_sequencing.jpg?w=1024" alt="developments_in_high_throughput_sequencing" width="1024" height="723" /></a></p>
<p><strong>Notable changes from the June 2014 edition</strong></p>
<ul>
<li>I added the Illumina HiSeq 4000</li>
<li>the HiSeq 2500 Rapid Run upgraded to 2x250 bp read length</li>
<li>PacBio upgraded to P6-C4 metrics</li>
<li>read numbers (but not the other metrics) for the full PacBio runs were updated, as they previously reflected those for single SMRTCells</li>
</ul>
<p><strong>But, where is the Oxford Nanopore MinION?</strong></p>
<p>The Oxford Nanopore MinION is a bit tricky. My metrics are based on company specifications that anyone can view from their website, for commercially released instruments and (chemistry) updates. The commercial release of the MinION seems now to have happened, as it was announced everyone can apply for the MinIon Access Program and will be accepted (barring some sanity screening they probably will do). But metrics for this instrument are a different matter. There are no company specs that I can find. Partly this is understandable, as read length, for example, is dependent on the input length of the sample (or library, rather). The other reason for the lack of specifications may be the <a href="https://www.nanoporetech.com/community/the-minion-access-programme-philosophy">Minion Access Programs philosophy</a> of Oxford Nanopore. It is the users that are discovering what the instrument can do, rather than the company telling the customer what to expect.</p>
<p>For my visualisation, this lack of specification causes a problem, as this makes it impossible to point to a source for where to place the MinION on the plot. Hence it is lacking from the figure above.</p>
<p>As a (temporary?) solution, here is, purely based on what I read in articles, both published scientific articles, preprints, and webpages, where the MinION more or less belongs. Note that I choose to represent it with a cloud rather than a single, solid datapoint...</p>
<p>&nbsp;</p>
<p><a href="https://flxlexblog.files.wordpress.com/2015/06/june_2015_with_minion1.jpg"><img class="alignnone size-large wp-image-682" src="{{ site.baseurl }}/assets/june_2015_with_minion1.jpg?w=1024" alt="June_2015_with_MinION" width="1024" height="723" /></a></p>
<p><strong>Some comments</strong></p>
<ul>
<li>note how close together the data points fall for the HiSeq 4000 and the HiSeq X</li>
<li>[EDIT] another new 'instrument' is the <a href="http://www.illumina.com/systems/hiseq-x-sequencing-system/x-five-system.html">HiSeq X Five</a>, but the single instrument is the same as for the HiSeq X Ten, and so now new datapoint was generated for the X Five</li>
<li>Illumina instruments clearly dominate the top middle part of the figure (between 100 and 300 bp, and 10 GB and 2 TB throughput)</li>
<li>the Complete Genomics/BGI <a href="www.completegenomics.com/revolocity/">Revolocity</a> is missing from the figure, also since there are no company specifications on the Complete Genomics website. There is as of yet not enough information to try to piece together what the metrics are, all that is known is that the read length is <a href="https://www.genomeweb.com/sequencing-technology/bgis-complete-genomics-launches-new-ngs-system-capable-sequencing-10k-genomes">28 bp paired end</a>.</li>
<li>[EDIT] a reader left <a href="https://flxlexblog.wordpress.com/2014/06/11/developments-in-next-generation-sequencing-june-2014-edition/#comment-2058">a comment</a> which I'd like to quote: "The Amersham/GE Healthcare MegaBACE 4500 was a 384-capillary instrument, with readlengths over 1000 bp. Yet, due to ABI’s better dominance in the market, the MB 4500 never had much penetration."</li>
<li>[EDIT] see also <a href="https://flxlexblog.wordpress.com/2014/06/11/developments-in-next-generation-sequencing-june-2014-edition/#comment-3226">this reader comment</a> on the history of the ABI 37* instruments. I've added 2002 as release date for the 3730xl as a result. It would be fun to try to dig up more information on the other instruments on the market before the 454 and Solexa ones came out...</li>
<li>as mentioned in the original blog post: some data was obtained by going to previous versions of company websites through the Internet Archive</li>
<li>I used full single-run specs with maximally stated throughput as available at the time of writing</li>
<li>sometimes, the total numbers of reads per full run and total bases obtained do not match up; for the figure, I always chose the reported throughput in bases</li>
<li>for Illumina, I chose to use the single-end read length, although the maximum throughput was based on the sum of all reads from a paired end run; I felt it unfair to double the read length for this platform for the figure</li>
</ul>
<p><strong>Availability</strong><br />
Data and figures are released under a CC0 license at <a href="http://figshare.com/articles/developments_in_NGS/100940">figshare</a>, with doi <a href="http://dx.doi.org/10.6084/m9.figshare.100940">10.6084/m9.figshare.100940</a>. I’ve also added the content to Github at https://github.com/lexnederbragt/developments-in-next-generation-sequencing.</p>
<p><strong>Disclaimer</strong><br />
As before: although I took utmost care in collecting the data, I may have gotten some of my numbers completely wrong, for which I apologise in advance; please help me correct any mistakes or omissions through leaving a comment, or sending me a pull request.</p>
<p><strong>Finally, the raw data</strong></p>
<table>
<thead>
<tr>
<th>Platform</th>
<th>Instrument</th>
<th>Year</th>
<th>Reads per run</th>
<th>Read length (mode or average)</th>
<th>Bases per run (gigabases)</th>
<th>Source</th>
</tr>
</thead>
<tbody>
<tr>
<td>ABI Sanger</td>
<td>3730xl</td>
<td>2002</td>
<td>96</td>
<td>800</td>
<td>0.0000768</td>
<td>0</td>
</tr>
<tr>
<td>454</td>
<td>GS20</td>
<td>2005</td>
<td>200000</td>
<td>100</td>
<td>0.02</td>
<td></td>
</tr>
<tr>
<td>454</td>
<td>GS FLX</td>
<td>2007</td>
<td>400000</td>
<td>250</td>
<td>0.1</td>
<td></td>
</tr>
<tr>
<td>454</td>
<td>GS FLX Titanium</td>
<td>2009</td>
<td>1000000</td>
<td>500</td>
<td>0.45</td>
<td></td>
</tr>
<tr>
<td>454</td>
<td>GS FLX+</td>
<td>2011</td>
<td>1000000</td>
<td>700</td>
<td>0.7</td>
<td>1</td>
</tr>
<tr>
<td>454</td>
<td>GS Junior</td>
<td>2010</td>
<td>100000</td>
<td>400</td>
<td>0.04</td>
<td>2</td>
</tr>
<tr>
<td>454</td>
<td>GS Junior+</td>
<td>2014</td>
<td>100000</td>
<td>700</td>
<td>0.07</td>
<td>16</td>
</tr>
<tr>
<td>IonTorrent</td>
<td>PGM 314 chip</td>
<td>2011</td>
<td>100000</td>
<td>100</td>
<td>0.01</td>
<td>3</td>
</tr>
<tr>
<td>IonTorrent</td>
<td>PGM 316 chip</td>
<td>2011</td>
<td>1000000</td>
<td>100</td>
<td>0.1</td>
<td>3</td>
</tr>
<tr>
<td>IonTorrent</td>
<td>PGM 318 chip</td>
<td>2011</td>
<td>5000000</td>
<td>100</td>
<td>0.5</td>
<td>3</td>
</tr>
<tr>
<td>IonTorrent</td>
<td>PGM 318 chip</td>
<td>2012</td>
<td>5000000</td>
<td>200</td>
<td>1</td>
<td>3</td>
</tr>
<tr>
<td>IonTorrent</td>
<td>PGM 318 chip V2</td>
<td>2013</td>
<td>5000000</td>
<td>400</td>
<td>2</td>
<td>12</td>
</tr>
<tr>
<td>IonTorrent</td>
<td>Proton PI</td>
<td>2012</td>
<td>50000000</td>
<td>200</td>
<td>10</td>
<td>4</td>
</tr>
<tr>
<td>Illumina (Solexa)</td>
<td>GA</td>
<td>2006</td>
<td>28000000</td>
<td>25</td>
<td>0.7</td>
<td></td>
</tr>
<tr>
<td>Illumina</td>
<td>GA</td>
<td>2008</td>
<td>28000000</td>
<td>35</td>
<td>1</td>
<td>5</td>
</tr>
<tr>
<td>Illumina</td>
<td>GA II</td>
<td>ND</td>
<td>100000000</td>
<td>50</td>
<td>5</td>
<td></td>
</tr>
<tr>
<td>Illumina</td>
<td>GAIIx</td>
<td>2009</td>
<td>440000000</td>
<td>75</td>
<td>33</td>
<td>6</td>
</tr>
<tr>
<td>Illumina</td>
<td>GAIIx</td>
<td>2011</td>
<td>640000000</td>
<td>75</td>
<td>48</td>
<td>7</td>
</tr>
<tr>
<td>Illumina</td>
<td>GAIIx</td>
<td>2012</td>
<td>640000000</td>
<td>150</td>
<td>95</td>
<td>8</td>
</tr>
<tr>
<td>Illumina</td>
<td>HiSeq 2000</td>
<td>2010</td>
<td>2000000000</td>
<td>100</td>
<td>200</td>
<td>9</td>
</tr>
<tr>
<td>Illumina</td>
<td>HiSeq 2000</td>
<td>2011</td>
<td>3000000000</td>
<td>100</td>
<td>600</td>
<td>10</td>
</tr>
<tr>
<td>Illumina</td>
<td>HiSeq 2000/2500</td>
<td>2014</td>
<td>4000000000</td>
<td>125</td>
<td>1000</td>
<td>17</td>
</tr>
<tr>
<td>Illumina</td>
<td>HiSeq 2500 RR</td>
<td>2012</td>
<td>600000000</td>
<td>150</td>
<td>180</td>
<td>13</td>
</tr>
<tr>
<td>Illumina</td>
<td>HiSeq 2500 RR</td>
<td>2014</td>
<td>600000000</td>
<td>250</td>
<td>300</td>
<td>13</td>
</tr>
<tr>
<td>Illumina</td>
<td>HiSeq 4000</td>
<td>2015</td>
<td>5000000000</td>
<td>150</td>
<td>1500</td>
<td>19</td>
</tr>
<tr>
<td>Illumina</td>
<td>HiSeq X</td>
<td>2014</td>
<td>6000000000</td>
<td>150</td>
<td>1800</td>
<td>18</td>
</tr>
<tr>
<td>Illumina</td>
<td>NextSeq 500</td>
<td>2014</td>
<td>400000000</td>
<td>150</td>
<td>120</td>
<td>14</td>
</tr>
<tr>
<td>Illumina</td>
<td>MiSeq</td>
<td>2011</td>
<td>30000000</td>
<td>150</td>
<td>4.5</td>
<td></td>
</tr>
<tr>
<td>Illumina</td>
<td>MiSeq</td>
<td>2012</td>
<td>30000000</td>
<td>250</td>
<td>8.5</td>
<td>11</td>
</tr>
<tr>
<td>Illumina</td>
<td>MiSeq</td>
<td>2013</td>
<td>30000000</td>
<td>300</td>
<td>15</td>
<td>14</td>
</tr>
<tr>
<td>SOLiD</td>
<td>1</td>
<td>2007</td>
<td>40000000</td>
<td>25</td>
<td>1</td>
<td></td>
</tr>
<tr>
<td>SOLiD</td>
<td>2</td>
<td>2008</td>
<td>115000000</td>
<td>35</td>
<td>4</td>
<td></td>
</tr>
<tr>
<td>SOLiD</td>
<td>3</td>
<td>2009</td>
<td>320000000</td>
<td>50</td>
<td>16</td>
<td></td>
</tr>
<tr>
<td>SOLiD</td>
<td>4</td>
<td>2010</td>
<td>2000000000</td>
<td>50</td>
<td>100</td>
<td></td>
</tr>
<tr>
<td>SOLiD</td>
<td>5500xl</td>
<td>2011</td>
<td>3000000000</td>
<td>60</td>
<td>180</td>
<td></td>
</tr>
<tr>
<td>SOLiD</td>
<td>5500xl W</td>
<td>2013</td>
<td>3000000000</td>
<td>75</td>
<td>320</td>
<td></td>
</tr>
<tr>
<td>PacBio</td>
<td>RS C1</td>
<td>2011</td>
<td>432000</td>
<td>1300</td>
<td>0.540</td>
<td></td>
</tr>
<tr>
<td>PacBio</td>
<td>RS C2</td>
<td>2012</td>
<td>432000</td>
<td>2500</td>
<td>1.080</td>
<td></td>
</tr>
<tr>
<td>PacBio</td>
<td>RS C2 XL</td>
<td>2012</td>
<td>432000</td>
<td>4300</td>
<td>1.858</td>
<td></td>
</tr>
<tr>
<td>PacBio</td>
<td>RS II C2 XL</td>
<td>2013</td>
<td>564000</td>
<td>4600</td>
<td>2.594</td>
<td>15</td>
</tr>
<tr>
<td>PacBio</td>
<td>RS II P5 C3</td>
<td>2014</td>
<td>528000</td>
<td>8500</td>
<td>4.500</td>
<td>15</td>
</tr>
<tr>
<td>PacBio</td>
<td>RS II P6 C4</td>
<td>2014</td>
<td>660000</td>
<td>13500</td>
<td>9.000</td>
<td>15</td>
</tr>
</tbody>
</table>
<p>[1] mode or average<br />
[2] Sources: see <a href="https://raw.githubusercontent.com/lexnederbragt/developments-in-next-generation-sequencing/master/sources.txt">this file</a> from the github repo.</p>
