---
layout: post
title: 'The Roche/454 Life Sciences GS FLX and GS Junior: an obituary'
date: 2016-10-21 14:43:19.000000000 +02:00
type: post
published: true
status: publish
categories:
- Next Generation Sequencing
tags:
- newbler
meta:
  _wpcom_is_markdown: '1'
  publicize_google_plus_url: https://plus.google.com/113347535998207715225/posts/XdxzNeKu5YS
  _rest_api_published: '1'
  _rest_api_client_id: '-1'
  _publicize_job_id: '28080819128'
  _publicize_done_6048082: '1'
  _wpas_done_6015510: '1'
author:
  login: lexnederbragt
  email: lex.nederbragt@bio.uio.no
  display_name: lexnederbragt
  first_name: ''
  last_name: ''
---
<p>You were a pioneer, the first successful 'next generation' (if you'll <a href="http://www.acgt.me/blog/2014/3/7/next-generation-sequencing-must-die">pardon the term</a>) commercially available sequencing platform in 2005. You just beat Solexa, but it was a fairly close call.</p>
<p>[caption width="631" id="attachment_803" align="alignnone"]<img class="alignnone size-full wp-image-803" src="{{ site.baseurl }}/assets/gsflx-uio1.png" alt="gsflx-uio1" width="631" height="474" /> The author (left) with colleagues showing off their 454 GS FLX[/caption]Your greatest accomplishment was to show that pyrosequencing, which was around for a while already, could be scaled up, both in terms of read length and parallelisation. You started the revolution in DNA sequencing, suddenly making large scale genomic projects available to labs that traditionally only could dream of been doing such projects at this scale.</p>
<p><!--more--></p>
<p>And then you were swallowed up by the giant, Roche diagnostics. Opinions will always be divided whether that was a good thing for you. Some say Roche, with its large sales network, enabled much more rapid spreading of your technology into research labs all over the world. Others will blame Roche for ultimately stifling innovation and progress.</p>
<p>As with any new (sequencing) technology, you had a challenge making sure researchers would be able to use the data they were getting. Luckily, you employed a bunch of smart bioinformaticians and programmers, leading to the newbler (gsAssembly, gsMapping) applications. I have been, and still am, a fan of this software. I have shared my enthusiasm by writing the <a href="https://contig.wordpress.com/table-of-contents/">'user-perspective manual' for newbler</a> (gsAssembler). I have <a href="https://flxlexblog.wordpress.com/2014/01/31/make-newbler-open-source/">unsuccessfully tried to make your owner open up this software</a>, asking you, on behalf of more than 200 users, to make it open source. Your superiors unfortunately <a href="https://flxlexblog.wordpress.com/2014/03/17/make-newbler-open-source-the-roche-response-and-the-future-of-newbler/">refused</a>. I will never give up hope that this still can become part of your legacy.</p>
<p>You grew fast. First, you were only able to give us 100 nucleotide long reads - much shorter than what we were used to from 'Sanger' sequencing, but significantly longer than your competitor, with their initial 25 or 35 nucleotide reads. It wasn't a very wise decision to have the expected output (20 million basepairs per run) as part of the name of your first instrument (GS20, Genome Sequencer 20) as you could have known your output would grow. The name of the updated instrument that came in 2007, GS FLX, was much better. Not only was it flexible in terms of run setup (2, 4, 8 or 16 lanes), it also allowed you to stick to the name once new chemistry and software updated pushed read length and output. Now you could do 250 bp and ultimately 450 Mbp per run. Later you managed to increase to 500 bp with Titanium chemistry, and even to around Sanger lengths, up to 1 kb, with GS FLX+ - even though this meant a slight modification to the instrument to enable larger supplies to be used.</p>
<p>You got a little brother, aptly named the GS Junior. It was at the time when 'desktop' instruments became the new thing for all platforms, enabling also smaller labs to purchase a sequencing instrument.</p>
<p>You enabled a lot of science. According to <a href="http://454.com/publications/all-publications.asp">your website</a>, there are 3082 peer-reviewed publications enabled by 454 Sequencing technology - but the most recent one listed there is from 2013 so the real number is probably quite a bit higher. Some scienctific highlights are</p>
<ul>
<li>the study presenting your technology in <a href="http://www.nature.com/nature/journal/v437/n7057/full/nature03959.html">2005</a></li>
<li>the presence of ammonia-oxidizing archaea (AOA) in soil in <a href="http://www.nature.com/nature/journal/v442/n7104/full/nature04983.html">2006</a></li>
<li>microbial diversity in the deep sea in <a href="http://www.pnas.org/content/103/32/12115.abstract">2006</a></li>
<li>the obesity-associated gut microbiome in mice in <a href="http://www.nature.com/nature/journal/v444/n7122/full/nature05414.html">2006</a></li>
<li>the diploid genome of James D. Watson in <a href="http://www.nature.com/nature/journal/v452/n7189/full/nature06884.html">2008</a></li>
<li>the sequencing of the Zebrafish antibody repertoire in <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3086368/">2009</a></li>
<li>a Neandertal Genome (also involving data from your competitor) in <a href="http://science.sciencemag.org/content/328/5979/710.full">2010</a></li>
</ul>
<p>I sometimes think you did not really reach full potential. Why not miniaturise even more, aim for even longer reads, make a new instrument that had ten, or a hundred times more output? We'll never know, now, what our R&amp;D crew were secretly working on. Your competitors surely did improve and increase throughput. Notably, one of your founders even <a href="https://en.wikipedia.org/wiki/Jonathan_Rothberg#Ion_Torrent">started from scratch</a>, reportedly aiming to 'do everything right' that you weren't - at least in their opinion. They managed partially, but due to strong competition from the giant of all giants in the sequencing world, are now pushed back into the niche market of targeted, gene-panel based projects.</p>
<p>You forced me to become a bioinformatician. Once I started to work with your data, they no longer fitted into excel, and I had to start using the command-line, and scripting. I am very glad this happened, it was a rewarding transition for me and made my research life much more interesting. At the University of Oslo, you lay the foundation of what later became the <a href="sequencing.uio.no">Norwegian Sequencing Centre</a>, a national sequencing technology service platform, which still is in operation today. We always have had a research-centred approach, using the technology we offer our users ourselves to make sure we understand it in and out. We used you to sequence several bacteria we were working on, and, an achievement I am still very proud of, to produce <a href="http://www.nature.com/nature/journal/v477/n7363/full/nature10342.html">the first version of the Atlantic cod genome in 2011</a>. That our lab was able to do this project was unthinkable before you and your competitors made sequencing much more accessible. We are still very grateful for all the help your owners have given us during the realisation of the project - we really did this together.</p>
<p>It was a sad day when it was<a href="https://www.genomeweb.com/sequencing/roche-shutting-down-454-sequencing-business"> announced you would be discontinued as a technology</a>. I guess it is a natural thing for new technology to become outdated and replaced (the exception maybe being the Sanger sequencing technology, which <a href="https://www.thermofisher.com/order/catalog/product/3730XL">still is around in the same form</a> as when you were launched). Our instruments are still in that lab, as we cannot bear to think dumping them - hopefully one of them can go to a museum.</p>
<p>I'll end with a big thank you. Thank you for pushing the boundaries, for your collaborative spirit, for the well-organised user meetings, for enabling all this science, and for turning me into a bioinformatician.</p>
<p>May you rest in peace and never be forgotten.</p>
<p>&nbsp;<br />
P.S. was the name of your mother company suggestive of the major sequencing error? "here we read 4, eh 5, no really 4 A's"</p>
