---
layout: post
title: Developments in high throughput sequencing – July 2016 edition
date: 2016-07-08 10:25:23.000000000 +02:00
type: post
published: true
status: publish
categories:
- Next Generation Sequencing
tags:
- '454'
- Illumina
- Ion Torrent
- oxford nanopore
- pacbio
meta:
  _wpcom_is_markdown: '1'
  _edit_last: '12058266'
  geo_public: '0'
  publicize_google_plus_url: https://plus.google.com/113347535998207715225/posts/FVQxmtjXsLL
  _publicize_job_id: '24583640484'
  _publicize_done_6048082: '1'
  _wpas_done_6015510: '1'
  _wpas_skip_6015510: '1'
author:
  login: lexnederbragt
  email: lex.nederbragt@bio.uio.no
  display_name: lexnederbragt
  first_name: ''
  last_name: ''
---
<p>This is the fifth edition of this visualisation, previous editions were in <a href="/2015/06/17/developments-in-high-throughput-sequencing-june-2015-edition/">June 2015</a>, <a href="/2014/06/11/developments-in-next-generation-sequencing-june-2014-edition/">June 2014</a>, <a href="/2013/10/01/developments-in-next-generation-sequencing-october-2013-edition/">October2013</a> and <a href="/2012/12/03/developments-in-next-generation-sequencing-a-visualisation/">December 2012</a>.</p>
<p>As before, full run throughput in gigabases (billion bases) is plotted against single-end read length for the different sequencing platforms, both on a log scale. Yes, I know a certain new instrument (different from last time) seems to be missing, hang on, I’m coming back to that…</p>
<p><a href="https://flxlexblog.files.wordpress.com/2016/07/developments_in_high_throughput_sequencing.jpg"><img class="aligncenter size-large wp-image-795" src="{{ site.baseurl }}/assets/developments_in_high_throughput_sequencing.jpg?w=1024" alt="developments_in_high_throughput_sequencing" width="1024" height="723" /></a></p>
<p><strong>Notable changes from the June 2015 edition</strong></p>
<ul>
<li>I added the Illumina MiniSeq</li>
<li>I added the Oxford Nanopore MinION. The read length for this instrument was based on the specifications for maximal output and number of reads from <a href="https://nanoporetech.com/products/specifications">the company's website</a>. The two data points represent 'regular' and 'fast' modes.</li>
<li>I added the IonTorrent S5 and S5XL. You may notice that the line for this instrument has a downward slope, this is due to the fact that the 400 bp reads are only available on the 520 and 530 chip, but not the higher throughput 540 chip, making the maximum throughput for this read length lower than for the 200 bp reads.</li>
</ul>
<p><!--more--></p>
<p><strong>But, where is the PacBio Sequel?</strong></p>
<p>Last year, Pacific Biosciences <a href="http://www.pacb.com/blog/introducing-the-sequel-system-the-scalable-platform-for-smrt-sequencing/">announced</a> the launch of the Sequel instrument, which according <a href="http://www.pacb.com/products-and-services/pacbio-systems/sequel/">their website</a> "delivers about 7X more reads with 1 million zero-mode waveguides (ZMWs) per SMRT Cell". However, there are currently no specifications for this instrument on the PacBio website. The <a href="http://www.pacb.com/wp-content/uploads/2015/09/Revolutionize-Genomics-with-SMRT-Sequencing.pdf">brochure about the Sequel System</a> has metrics of a typical run, but these are from data obtained on the PacBio RS II, <em>not</em> the Sequel. I appreciate it is not easy to come up with metrics for this type of sequencing technology, where so much depends on the sample and library types, but I still need them in order to make my visualisation. Thus, until the company adds specifications for their new instrument to their website (or brochures), I reluctantly cannot add it to my visualisation. The next best thing PacBio could do would be releasing some data (and the company <a href="https://github.com/PacificBiosciences/DevNet/wiki/Datasets">has an excellent history of doing so</a>). If these specifications or data appear soon, I'll update the illustration and this blog post.</p>
<p><strong>Some comments</strong></p>
<ul>
<li>look how crowded the 150-300 bp, 1-10 Gbp area of the plot has become...</li>
<li>I finally switched to using python for generating the 'bare' plot (without the images of the instruments), see the Jupyter notebook in the <a href="https://github.com/lexnederbragt/developments-in-next-generation-sequencing">git repo</a>.</li>
<li>as mentioned in the original blog post: some data was obtained by going to previous versions of company websites through the Internet Archive</li>
<li>I used full single-run specs with maximally stated throughput as available at the time of writing</li>
<li>sometimes, the total numbers of reads per full run and total bases obtained do not match up; for the figure, I always chose the reported throughput in bases</li>
<li>for Illumina, I chose to use the single-end read length, although the maximum throughput was based on the sum of all reads from a paired end run; I felt it unfair to double the read length for this platform for the figure</li>
</ul>
<p><strong>Availability</strong></p>
<p>Data and figures are released under a CC BY license at <a href="http://figshare.com/articles/developments_in_NGS/100940">figshare</a>, with doi <a href="http://dx.doi.org/10.6084/m9.figshare.100940">10.6084/m9.figshare.100940</a>. I’ve also added the content to Github <a href="https://github.com/lexnederbragt/developments-in-next-generation-sequencing">at this repository </a>.</p>
<p><strong>Disclaimer</strong></p>
<p>Data are based on metrics available from instrument providers websites, sources listed in the table can be found <a href="https://github.com/lexnederbragt/developments-in-next-generation-sequencing/blob/master/sources.txt">here</a>. As before: although I took utmost care in collecting the data, I may have gotten some of my numbers completely wrong, for which I apologise in advance; please help me correct any mistakes or omissions through leaving a comment (or sending me a pull request).</p>
<p><strong>Finally, the raw data</strong></p>
<table>
<thead>
<tr>
<th>Platform</th>
<th>Instrument</th>
<th>Label</th>
<th>Year</th>
<th>Reads per run</th>
<th>Read length (mode or average)</th>
<th>Bases per run (gigabases)</th>
<th><a href="https://github.com/lexnederbragt/developments-in-next-generation-sequencing/blob/master/sources.txt">Source</a></th>
</tr>
</thead>
<tbody>
<tr>
<td>ABI Sanger</td>
<td>3730xl</td>
<td>ABI Sanger 3730xl</td>
<td>2002</td>
<td>96</td>
<td>800</td>
<td>0.0000768</td>
<td>0</td>
</tr>
<tr>
<td>454</td>
<td>GS20</td>
<td>454 GS FLX</td>
<td>2005</td>
<td>200000</td>
<td>100</td>
<td>0.02</td>
<td></td>
</tr>
<tr>
<td>454</td>
<td>GS FLX</td>
<td>454 GS FLX</td>
<td>2007</td>
<td>400000</td>
<td>250</td>
<td>0.1</td>
<td></td>
</tr>
<tr>
<td>454</td>
<td>GS FLX Titanium</td>
<td>454 GS FLX</td>
<td>2009</td>
<td>1000000</td>
<td>500</td>
<td>0.45</td>
<td></td>
</tr>
<tr>
<td>454</td>
<td>GS FLX+</td>
<td>454 GS FLX</td>
<td>2011</td>
<td>1000000</td>
<td>700</td>
<td>0.7</td>
<td>1</td>
</tr>
<tr>
<td>454</td>
<td>GS Junior</td>
<td>454 GS Junior</td>
<td>2010</td>
<td>100000</td>
<td>400</td>
<td>0.04</td>
<td>2</td>
</tr>
<tr>
<td>454</td>
<td>GS Junior+</td>
<td>454 GS Junior</td>
<td>2014</td>
<td>100000</td>
<td>700</td>
<td>0.07</td>
<td>16</td>
</tr>
<tr>
<td>Illumina (Solexa)</td>
<td>GA</td>
<td>Illumina GA</td>
<td>2006</td>
<td>28000000</td>
<td>25</td>
<td>0.7</td>
<td></td>
</tr>
<tr>
<td>Illumina</td>
<td>GA</td>
<td>Illumina GA</td>
<td>2008</td>
<td>28000000</td>
<td>35</td>
<td>1</td>
<td>5</td>
</tr>
<tr>
<td>Illumina</td>
<td>GA II</td>
<td>Illumina GA</td>
<td>ND</td>
<td>100000000</td>
<td>50</td>
<td>5</td>
<td></td>
</tr>
<tr>
<td>Illumina</td>
<td>GAIIx</td>
<td>Illumina GA</td>
<td>2009</td>
<td>440000000</td>
<td>75</td>
<td>33</td>
<td>6</td>
</tr>
<tr>
<td>Illumina</td>
<td>GAIIx</td>
<td>Illumina GA</td>
<td>2011</td>
<td>640000000</td>
<td>75</td>
<td>48</td>
<td>7</td>
</tr>
<tr>
<td>Illumina</td>
<td>GAIIx</td>
<td>Illumina GA</td>
<td>2012</td>
<td>640000000</td>
<td>150</td>
<td>95</td>
<td>8</td>
</tr>
<tr>
<td>Illumina</td>
<td>HiSeq 2000</td>
<td>Illumina HiSeq 2000/2500</td>
<td>2010</td>
<td>2000000000</td>
<td>100</td>
<td>200</td>
<td>9</td>
</tr>
<tr>
<td>Illumina</td>
<td>HiSeq 2000</td>
<td>Illumina HiSeq 2000/2500</td>
<td>2011</td>
<td>3000000000</td>
<td>100</td>
<td>600</td>
<td>10</td>
</tr>
<tr>
<td>Illumina</td>
<td>HiSeq 2000/2500</td>
<td>Illumina HiSeq 2000/2500</td>
<td>2014</td>
<td>4000000000</td>
<td>125</td>
<td>1000</td>
<td>17</td>
</tr>
<tr>
<td>Illumina</td>
<td>HiSeq 2500 RR</td>
<td>Illumina HiSeq 2500 RR</td>
<td>2012</td>
<td>600000000</td>
<td>150</td>
<td>180</td>
<td>13</td>
</tr>
<tr>
<td>Illumina</td>
<td>HiSeq 2500 RR</td>
<td>Illumina HiSeq 2500 RR</td>
<td>2014</td>
<td>600000000</td>
<td>250</td>
<td>300</td>
<td>13</td>
</tr>
<tr>
<td>Illumina</td>
<td>HiSeq 4000</td>
<td>Illumina HiSeq 4000</td>
<td>2015</td>
<td>5000000000</td>
<td>150</td>
<td>1500</td>
<td>19</td>
</tr>
<tr>
<td>Illumina</td>
<td>HiSeq X</td>
<td>Illumina HiSeq X</td>
<td>2014</td>
<td>6000000000</td>
<td>150</td>
<td>1800</td>
<td>18</td>
</tr>
<tr>
<td>Illumina</td>
<td>NextSeq 500</td>
<td>Illumina NextSeq 500</td>
<td>2014</td>
<td>400000000</td>
<td>150</td>
<td>120</td>
<td>14</td>
</tr>
<tr>
<td>Illumina</td>
<td>MiSeq</td>
<td>Illumina MiSeq</td>
<td>2011</td>
<td>30000000</td>
<td>150</td>
<td>4.5</td>
<td></td>
</tr>
<tr>
<td>Illumina</td>
<td>MiSeq</td>
<td>Illumina MiSeq</td>
<td>2012</td>
<td>30000000</td>
<td>250</td>
<td>8.5</td>
<td>11</td>
</tr>
<tr>
<td>Illumina</td>
<td>MiSeq</td>
<td>Illumina MiSeq</td>
<td>2013</td>
<td>30000000</td>
<td>300</td>
<td>15</td>
<td>14</td>
</tr>
<tr>
<td>Illumina</td>
<td>MiniSeq</td>
<td>Illumina MiniSeq</td>
<td>2016</td>
<td>25000000</td>
<td>150</td>
<td>7.5</td>
<td>20</td>
</tr>
<tr>
<td>SOLiD</td>
<td>1</td>
<td>SOLiD</td>
<td>2007</td>
<td>40000000</td>
<td>25</td>
<td>1</td>
<td></td>
</tr>
<tr>
<td>SOLiD</td>
<td>2</td>
<td>SOLiD</td>
<td>2008</td>
<td>115000000</td>
<td>35</td>
<td>4</td>
<td></td>
</tr>
<tr>
<td>SOLiD</td>
<td>3</td>
<td>SOLiD</td>
<td>2009</td>
<td>320000000</td>
<td>50</td>
<td>16</td>
<td></td>
</tr>
<tr>
<td>SOLiD</td>
<td>4</td>
<td>SOLiD</td>
<td>2010</td>
<td>2000000000</td>
<td>50</td>
<td>100</td>
<td></td>
</tr>
<tr>
<td>SOLiD</td>
<td>5500xl</td>
<td>SOLiD</td>
<td>2011</td>
<td>3000000000</td>
<td>60</td>
<td>180</td>
<td></td>
</tr>
<tr>
<td>SOLiD</td>
<td>5500xl W</td>
<td>SOLiD</td>
<td>2013</td>
<td>3000000000</td>
<td>75</td>
<td>320</td>
<td></td>
</tr>
<tr>
<td>IonTorrent</td>
<td>PGM 314 chip</td>
<td>IonTorrent PGM</td>
<td>2011</td>
<td>100000</td>
<td>100</td>
<td>0.01</td>
<td>3</td>
</tr>
<tr>
<td>IonTorrent</td>
<td>PGM 316 chip</td>
<td>IonTorrent PGM</td>
<td>2011</td>
<td>1000000</td>
<td>100</td>
<td>0.1</td>
<td>3</td>
</tr>
<tr>
<td>IonTorrent</td>
<td>PGM 318 chip</td>
<td>IonTorrent PGM</td>
<td>2011</td>
<td>5000000</td>
<td>100</td>
<td>0.5</td>
<td>3</td>
</tr>
<tr>
<td>IonTorrent</td>
<td>PGM 318 chip</td>
<td>IonTorrent PGM</td>
<td>2012</td>
<td>5000000</td>
<td>200</td>
<td>1</td>
<td>3</td>
</tr>
<tr>
<td>IonTorrent</td>
<td>PGM 318 chip V2</td>
<td>IonTorrent PGM</td>
<td>2013</td>
<td>5000000</td>
<td>400</td>
<td>2</td>
<td>12</td>
</tr>
<tr>
<td>IonTorrent</td>
<td>Proton PI</td>
<td>IonTorrent Proton</td>
<td>2012</td>
<td>50000000</td>
<td>200</td>
<td>10</td>
<td>4</td>
</tr>
<tr>
<td>IonTorrent</td>
<td>Ion S5/S5XL 530 chip</td>
<td>IonTorrent S5/S5XL</td>
<td>2015</td>
<td>20000000</td>
<td>400</td>
<td>8</td>
<td>21</td>
</tr>
<tr>
<td>IonTorrent</td>
<td>Ion S5/S5XL 540 chip</td>
<td>IonTorrent S5/S5XL</td>
<td>2015</td>
<td>75000000</td>
<td>200</td>
<td>15</td>
<td>21</td>
</tr>
<tr>
<td>PacBio</td>
<td>RS C1</td>
<td>PacBio RS</td>
<td>2011</td>
<td>432000</td>
<td>1300</td>
<td>0.540</td>
<td></td>
</tr>
<tr>
<td>PacBio</td>
<td>RS C2</td>
<td>PacBio RS</td>
<td>2012</td>
<td>432000</td>
<td>2500</td>
<td>1.080</td>
<td></td>
</tr>
<tr>
<td>PacBio</td>
<td>RS C2 XL</td>
<td>PacBio RS</td>
<td>2012</td>
<td>432000</td>
<td>4300</td>
<td>1.858</td>
<td></td>
</tr>
<tr>
<td>PacBio</td>
<td>RS II C2 XL</td>
<td>PacBio RS</td>
<td>2013</td>
<td>564000</td>
<td>4600</td>
<td>2.594</td>
<td>15</td>
</tr>
<tr>
<td>PacBio</td>
<td>RS II P5 C3</td>
<td>PacBio RS</td>
<td>2014</td>
<td>528000</td>
<td>8500</td>
<td>4.500</td>
<td>15</td>
</tr>
<tr>
<td>PacBio</td>
<td>RS II P6 C4</td>
<td>PacBio RS</td>
<td>2014</td>
<td>660000</td>
<td>13500</td>
<td>12.000</td>
<td>15</td>
</tr>
<tr>
<td>Oxford Nanopore</td>
<td>MinION Mk1</td>
<td>Oxford Nanopore MinION Mk1</td>
<td>2015</td>
<td>2200000</td>
<td>9545</td>
<td>21</td>
<td>22</td>
</tr>
<tr>
<td>Oxford Nanopore</td>
<td>MinION Mk1 fast</td>
<td>Oxford Nanopore MinION Mk1</td>
<td>2015</td>
<td>4400000</td>
<td>9545</td>
<td>42</td>
<td>22</td>
</tr>
</tbody>
</table>
