---
layout: post
title: De novo genome assembly - before and after
date: 
type: post
published: false
status: draft
categories:
- Bioinformatics
tags: []
meta:
  _edit_last: '12058266'
author:
  login: lexnederbragt
  email: lex.nederbragt@bio.uio.no
  display_name: lexnederbragt
  first_name: ''
  last_name: ''
---
<p>De novo genome assembly is all about assembly programs, right? Well... The recent assembly-competeion papers already pointed out the value of trimming reads, and error-correcting them, see <a href="http://ccb.jhu.edu/gage_b/">GAGE-B</a> and <a href="http://assemblathon.org/">Assemblathon</a>. But there is much more to assembly than this.</p>
<p>With this post, I will list a few of the things one can (should?) do before, and after the actual process of performing the assembly. There are the following limitations:</p>
<ul>
<li>I do no treat preparing samples, deciding on a strategy (library type, sequencing platform and type, etc.)</li>
<li>I will say nothing about the actual assembly programs and this post should be seen as independent of such programs</li>
<li>the approaches I use only make use of the read data available to you, and the assembly results (no optical maps, fosmids, RNA-Seq data etc)</li>
<li>No reference is needed for these approaches - which is the point of this whole post</li>
<li>This list is by no means complete, and tools listed are a selection of those availbe - which tools I list should not be seen as an endorsement - I just happen to know (of) them.</li>
<li>I repeat, no <em>reference is needed </em></li>
</ul>
<ol>
<li>Before assembly</li>
</ol>
<p><span style="text-decoration:underline;">Quality control of your reads<br />
</span>Check quantity, quality (average per position, average per read), GC%, overrepresented subsequences, presence of adaptors/barcodes</p>
<ul>
<li><a href="http://www.bioinformatics.babraham.ac.uk/projects/fastqc/">fastqc</a></li>
<li><a href="http://prinseq.sourceforge.net/">prinseq</a></li>
<li>see <a href="http://seqanswers.com/wiki/Sequencing_Quality_Control">Seqwiki : Sequencing_Quality_Control</a></li>
</ul>
<p><span style="text-decoration:underline;">Genome and read characteristics<br />
</span>Estimate genome size, repeat content, polymorphism level, coverage.</p>
<p>preqc (SGA)</p>
<p><a href="https://github.com/wltrimbl/kmerspectrumanalyzer">Kmerspectrumanalyzer</a><br />
BGI's</p>
<p>For de Bruijn graphs: estimate kmer to use<br />
preqc (SGA)<br />
kmergenie<br />
others?</p>
<p>Data grooming<br />
adapter trimming<br />
quality trimming<br />
format conversion (e.g. from outie to innie, change read identifier, fastq version, splitting 454/ion mates)</p>
<p>Data reduction, improvement<br />
error-correcting reads<br />
merge pairs (flash/cope)<br />
lengthen reads (AFR-PE)<br />
digital normalisation</p>
<p>Assembly<br />
building contigs<br />
scaffolding</p>
<p>After assembly<br />
Basic metrics<br />
assemblathon metrics script<br />
others?</p>
<p>Assembly improvement<br />
gap closing<br />
recalling bases (iCORN, Quiver)<br />
correction (REAPR etc)</p>
<p>Assembly merging/reconciliation<br />
GAM<br />
Mike Scathz tool?<br />
CISA (Contig Integrator for Sequence Assembly)<br />
MIX (https://github.com/cbib/MIX)</p>
<p>Assembly QC/validation<br />
REAPR<br />
ALE<br />
FRCbam<br />
linkage map<br />
optical map<br />
compare to transcriptome data</p>
<p>Visualisation<br />
browser</p>
<p>Assembly comparison (likelihood)<br />
CGAL</p>
<p>Comparison to a reference<br />
Mauve contig mover<br />
Quast<br />
GAGE pipeline</p>
